---
title: August 2013 Galaxy Update
---
<div class='left'><a href='/src/galaxy-updates/index.md'><img src="/src/images/logos/GalaxyUpdate200.png" alt="Galaxy Updates" width=150 /></a></div>

Welcome to the August 2013 *[Galaxy Update](/src/galaxy-updates/index.md)*, a monthly summary of what is going on in the Galaxy community. *[Galaxy Updates](/src/galaxy-updates/index.md)* complement the *[Galaxy Development News Briefs](/src/docs/index.md)* which accompany new Galaxy releases and focus on Galaxy code updates.

<br />

<div class='center'><a href='/src/events/gcc2013/index.md'><img src="/src/images/logos/GCC2013Logo400.png" alt="2013 Galaxy Community Conference (GCC2013)" width="300" /></a> </div>

# GCC2013 Report

The [2013 Galaxy Community Conference (GCC2013)](/src/events/gcc2013/index.md) was held a month ago in Oslo Norway, at the [University of Oslo](http://uio.no).  [Video of all accepted talks](/src/events/gcc2013/program/index.md#day-1-main-meeting-1-july), the [slides for most accepted](/src/events/gcc2013/program/index.md#day-1-main-meeting-1-july) and [lightning talks](/src/events/gcc2013/lightning/index.md), [handouts and slides for all Training Day sessions](http://wiki.galaxyproject.org/Events/GCC2013/TrainingDay), and [most posters](/src/events/gcc2013/abstracts/index.md#poster-abstracts) are now available online.

GCC2013 was, by all accounts, a tremendous success.  This was the largest GCC ever held, and featured [poster sessions](/src/events/gcc2013/abstracts/index.md#poster-abstracts) and [Birds of a Feather gatherings](/src/events/gcc2013/bof/index.md) for the first time.  The evening [gatherings at the pub](/src/events/gcc2013/program/index.md#escape-to-the-pub) and the conference dinner at the historic Sporten venue, high above Oslo, also proved quite popular.

In addition to the videos and presentation materials, there are also several meeting reports in a wide variety of presentation styles:

* [#usegalaxy lights up northern skies](http://blogs.biomedcentral.com/gigablog/2013/07/09/usegalaxy-lights-up-northern-skies/): Meeting report by [Scott Edmunds](http://blogs.biomedcentral.com/gigablog/author/scottedmunds/) in [GigaBlog](http://blogs.biomedcentral.com/gigablog).
* [GCC2013 Report](https://depot.galaxyproject.org/hub/attachments/documents/presentations/gcc2013/NBICSummary.pdf) - an [NBIC](http://nbic.nl/) centric review of the meeting by Leon Mei.
* UiO Rector's Blog: [Reproduserbarhet i livsvitenskapene](http://bit.ly/12GQpvC) 
* [Conference Tweets at ChirpStory](http://chirpstory.com/li/93570)
* [Storify of conference Tweets](http://storify.com/search?q=GCC2013)
* [Photos](/src/events/gcc2013/photos/index.md) from the meeting (please add your own)

<br />

# New Public Servers

Two new servers joined [the list of over 30 publicly accessible Galaxy servers](/src/use/index.md) in July.

## PredPharmTox

<div class='left'><a href='http://galaxy.predpharmtox.org/'><img src="/src/use/archive/predpharmtox-logo.png" alt="PredPharmTox" width="200" /></a></div>

[PredPharmTox](http://galaxy.predpharmtox.org/) implements the [QuantMap tools](http://pubs.acs.org/doi/abs/10.1021/ci200429f) for grouping chemicals by biological activity. See "[Automated QuantMap for rapid quantitative molecular network topology analysis](http://bioinformatics.oxfordjournals.org/content/early/2013/07/04/bioinformatics.btt390.full.pdf)" by Schaal1, *et al.*, *[Bioinformatics](http://bioinformatics.oxfordjournals.org/)* (2013), for more.

## QBRC Galaxy

<div class='right'><a href='http://www.galaxy.qbrc.org/'><img src="/src/use/archive/qbrc-galaxy/qbrc-logo.jpg" alt="QBRC Galaxy" width="200" /></a></div>

The [QBRC Galaxy Server](http://www.galaxy.qbrc.org/) hosts custom tools developed by members of the [Quantitative Biomedical Research Center @ University of Texas Southwestern](http://www.qbrc.swmed.edu/) for statistical analysis of various High Throughput Sequencing experiments.

Features
* [SbacHTS: Spatial Background Noise Correction for HighThroughput RNAi Screening](http://bioinformatics.oxfordjournals.org/content/early/2013/06/28/bioinformatics.btt358.abstract)   
* [DecoRNAI](http://galaxy.qbrc.org/static/DecoRNAi_Manual.pdf) (deconvolution analysis of RNAi screening data)
* [MiClip](http://galaxy.qbrc.org/static/galaxy_manual.pdf) "a novel model-based approach to identify high-confidence protein-RNA binding sites in CLIP-Seq datasets"

The QBRC galaxy server tools are constantly being updated. Please contact authors for more details regarding use.

# New Papers

<table>
  <tr>
    <th> # </th>
    <th> Tag </th>
    <td rowspan=6 style=" border: none;"> &nbsp;&nbsp; </td>
    <th> # </th>
    <th> Tag </th>
    <td rowspan=4 style=" border: none;"> &nbsp;&nbsp; </td>
    <th> # </th>
    <th> Tag </th>
  </tr>
  <tr>
    <td style=" text-align: right;"> 22 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/methods'>methods</a> </td>
    <td style=" text-align: right;"> 4 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/cloud'>cloud</a> </td>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/project'>Project</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 15 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/workbench'>workbench</a> </td>
    <td style=" text-align: right;"> 3 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/usepublic'>usepublic</a> </td>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/reproducibility'>reproducibility</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 7 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/tools'>tools</a> </td>
    <td style=" text-align: right;"> 3 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/uselocal'>uselocal</a> </td>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/uselocal'>usecloud</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 7 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/usemain'>usemain</a> </td>
    <td style=" text-align: right;"> 3 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/shared'>shared</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 5 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/isgalaxy'>isgalaxy</a> </td>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/refpublic'>refpublic</a> </td>
  </tr>
</table>



47 new papers were added to the [Galaxy CiteULike Group](http://www.citeulike.org/group/16008/) in July. These papers may be particularly interesting to the Galaxy community
* "[A sustainable national gateway for biological computation](https://doi.org/10.1145/2484762.2484817)" by [James Taylor](/src/people/james-taylor/index.md), [Anton Nekrutenko](/src/people/anton/index.md), [Nate Coraor](/src/people/nate/index.md), *et al.* In *Proceedings of the Conference on Extreme Science and Engineering Discovery Environment: Gateway to Discovery* (2013)
* "[Cloud-based uniform ChIP-Seq processing tools for modENCODE and ENCODE](https://doi.org/10.1186/1471-2164-14-494)" by Quang Trinh, Fei Y. Jen, Ziru Zhou, *et al.*, *BMC Genomics,* Vol. 14, No. 1. (2013), 494
* "[The Banana Genome Hub](https://doi.org/10.1093/database/bat035)" by Droc, *et al.*, *Database*, Vol. 2013 (01 January 2013), 

<br />

# SlipStream Appliance: Galaxy Edition Announced

<div class='left'><a href='http://www.bioteam.net/slipstream/galaxy-edition'><img src="/src/images/logos/SlipStreamApplianceLogo.png" alt="SlipStream Appliance - Galaxy Edition" width="200" /></a></div>

[BioTeam](http://bioteam.net) has [announced](/src/news/slip-stream-appliance/index.md) the [SlipStream Appliance: Galaxy Edition](http://www.bioteam.net/slipstream/galaxy-edition) â€“ "a high-performance, server class device pre-loaded with a fully operational Galaxy analysis platform." It is a plug-and-play solution, "intended to make Galaxy more accessible by relieving some of the burdens users may experience with other methods of accessing the platform." See the [product page](http://www.bioteam.net/slipstream/galaxy-edition) for more details.

<br />

# New Galaxy Proteomics Mailing List

<div class='right'><a href='https://lists.galaxyproject.org/lists/galaxy-proteomics.lists.galaxyproject.org/'><img src="/src/images/logos/MailmanLogoSmall.png" alt="Galaxy Proteomics Mailing List"  /></a></div>

The Galaxy community now has a [mailing list dedicated to proteomics](https://lists.galaxyproject.org/lists/galaxy-proteomics.lists.galaxyproject.org/). If you're using Galaxy to do proteomics, or if you are thinking of doing so then [please subscribe](https://lists.galaxyproject.org/lists/galaxy-proteomics.lists.galaxyproject.org/).  In case you missed it, there is also a proteomics wiki page. 

See [the complete list of mailing lists](/src/mailing-lists/index.md) for more information on this and other Galaxy mailing lists, including links to searchable archives of all Galaxy lists.

Many thanks to Ira Cooke and [John Chilton](/src/people/john-chilton/index.md) for suggesting this and taking the lead in gathering the Galaxy Proteomics community.

<br />

# Who's Hiring

<div class='right'><a href='/src/galaxy-is-hiring/index.md'><img src="/src/galaxy-is-hiring/GalaxyIsHiringWordCloud2.png" alt="Please Help! Yes you!" width="200" /></a></div>

The Galaxy is expanding! Please help it grow.
* The Netherlands Institute of Ecology (NIOO-KNAW) [Bioinformatician opening](http://bit.ly/15nT1nB) 
* [PhD or postdoc position available](http://bit.ly/15Iz3nL) at [Laboratory of Computational Biology](http://med.kuleuven.be/lcb/), University of Leuven 
* [Bioinformatics job vacancy at Rothamsted Research](http://www.rothamsted.ac.uk/Content-Section=Careers&Page=Vacancies.html#1272), UK.  Use Galaxy for delivery of data analysis resources; new tool integration, ... **Closing date 4th August.**
* **The [Galaxy Project is hiring software engineers and post-docs](/src/galaxy-is-hiring/index.md)**.
* [Sr Bioinformatics Specialist, Tufts University](http://bit.ly/1bOLLBd), Boston MA.
* [Senior Developer](http://www.nature.com/naturejobs/science/jobs/334107-senior-developer-stem-cell-bioinformatics-core), Stem Cell Bioinformatics Core, Sage Bionetworks, Seattle, WA, United States
* [Bioinformatics Support Group Leader](http://bit.ly/VkRtrb) @ LSU

Got a Galaxy-related opening? Send it to outreach@galaxyproject.org and we'll put it in the [Galaxy News feed](/src/news/index.md) and include it in next month's [update](/src/galaxy-updates/index.md).

<br />

# Events

<div class='right'>
<a href='http://www.iscb.org/ismbeccb2013'><img src="/src/images/logos/ISMB2013Logo205.png" alt="ISMB/ECCB 2013" width="80" /></a>
</div>
<div class='right'>
<a href='http://www.open-bio.org/wiki/BOSC_2013'><img src="/src/images/logos/BOSC_logo.png" alt="BOSC 2013" width="60" /></a><br /><a href='http://igenomed2.stanford.edu/Proteomics2013_mssig/'><img src="/src/images/logos/MSSIG2013Logo.jpg" alt="MS SIG 2013: Beyond the Proteomics" width="60" /></a><br />
</div>

## ISMB / ECCB / BOSC / MS SIG 2013

Galaxy had a strong presence at [ISMB / ECCB 2013](http://www.iscb.org/ismbeccb2013), [BOSC](http://www.open-bio.org/wiki/BOSC_2013), and [MS SIG](http://igenomed2.stanford.edu/Proteomics2013_mssig/) in Berlin this year.  [Slides for 15 talks, posters, workshops](/src/events/ismb2013/index.md) related to Galaxy are [now available online](/src/events/ismb2013/index.md).  Also see the [summary from Bioinformatics Open Science Codefest 2013: Tools, infrastructure, standards and visualization](http://j.mp/13nJZ9m).  

<br />

## Other Upcoming Events

<div class='center'>
<a href='https://training.bioinformatics.ucdavis.edu/2013/07/02/the-2013-rna-seq-workshop/'><img src="/src/images/logos/UCDavisGenomeCenterLogo.jpg" alt="RNA-Seq Workshop: From Pipette to P-value! @ UC Davis Bioinformatics Core" height="60" /></a>
<a href='http://www.rcpa.edu.au/Continuing/CalendarOfEvents/September2013.htm'><img src="/src/images/logos/RCPALogo.jpg" alt="The Genomic Bioinformatics Workshop" height="60" /></a>
<a href='http://individualizingmedicineconference.mayo.edu/schedule/'><img src="/src/images/logos/MayoClinicSmall.png" alt="Individualizing Medicine Conference" height="60" /></a> &nbsp;
<a href='http://www.beyond-the-genome.com/program/'><img src="/src/images/logos/BeyondTheGenome2013.png" alt="Beyond the Genome" height="60" /></a> &nbsp;
<a href='http://bit.ly/153avr8'><img src="/src/images/logos/OrioneNGSCourse.png" alt="Analisi dati Next Generation Sequencing in Galaxy: exome, RNA-Seq, metagenomica" height="65" /></a> &nbsp;
<a href='http://bioinfo.genotoul.fr/index.php?id=34&tx_seminars_pi1%5BshowUid%5D=43'><img src="/src/images/logos/GennoToulLogo.png" alt="Galaxy Training Days" height="60" /></a>
</div>

<div class='right'><a href='http://bit.ly/gxycal'><img src="/src/images/icons/CalendarIcon.gif" /></a></div> See the [Galaxy Events Google Calendar](http://bit.ly/gxycal) for details on these and other events.

<br />

<table>
  <tr class="th" >
    <th> Date </th>
    <th> Topic/Event </th>
    <th> Venue/Location </th>
    <th> Contact </th>
  </tr>
  <tr>
    <th> September 9-11 </th>
    <td> <em><a href='https://training.bioinformatics.ucdavis.edu/2013/07/02/the-2013-rna-seq-workshop/'>RNA-Seq Workshop: From Pipette to P-value!</a></em> </td>
    <td rowspan=1> <a href='http://bioinformatics.ucdavis.edu/'>UC Davis Bioinformatics Core</a> Davis, California, United States </td>
    <td rowspan=1> <a href="mailto:ucdbio AT gmail DOT com">Email</a> </td>
  </tr>
  <tr>
    <th> September&nbsp;28 - October&nbsp;1 </th>
    <td> <em>Galaxy Workshop</em> </td>
    <td> <a href='http://www.rcpa.edu.au/Continuing/CalendarOfEvents/September2013.htm'>The Genomic Bioinformatics Workshop</a>, Sydney, Australia </td>
    <td> <a href='/src/people/jennifer-jackson/index.md'>Jennifer Jackson</a>, <a href='/src/people/fubar/index.md'>Ross Lazarus</a> </td>
  </tr>
  <tr>
    <th> September 30 - October 2 </th>
    <td> <em>TBA</em>, part of the <em><a href='http://individualizingmedicineconference.mayo.edu/schedule/'>Cancer Care</a></em> session </td>
    <td> <a href='http://individualizingmedicineconference.mayo.edu/'>Individualizing Medicine Conference</a>, <a href='http://mayo.eduy/'>Mayo Clinic</a>, Rochester, Minnesota, United States </td>
    <td> <a href='/src/people/james-taylor/index.md'>James Taylor</a> </td>
  </tr>
  <tr>
    <th> October 1-3 </th>
    <td> <em><a href='http://www.beyond-the-genome.com/program/'>Galaxy</a></em> </td>
    <td> <strong><a href='http://www.beyond-the-genome.com/'>Beyond the Genome 2013</a></strong>,San Francisco, California, United States </td>
    <td> <a href='/src/people/jeremy-goecks/index.md'>Jeremy Goecks</a> </td>
  </tr>
  <tr>
    <th rowspan=2> October 7-8 </th>
    <td> <em><a href='https://www.gtcbio.com/conference/ngseurope-agenda#Day 1 -BIO'>TBD</a></em></td>
    <td rowspan=2> <a href='http://www.gtcbio.com/conference/ngseurope-overview'>NGS & Bioinformatics Summit Europe</a> </td>
    <td> <a href='/src/people/anton/index.md'>Anton Nektutenko</a> </td>
  </tr>
  <tr>
    <td> <em><a href='https://www.gtcbio.com/conference/ngseurope-agenda#Day 1 -BIO'>Using Galaxy to Provide a NGS Analysis Platform</a> </em> </td>
    <td> <a href='/src/people/hansrudolf-hotz/index.md'>Hans-Rudolf Hotz</a> </td>
  </tr>
  <tr>
    <th> October 8-11 </th>
    <td> <em><a href='http://bit.ly/153avr8'>Analisi dati Next Generation Sequencing in Galaxy: exome, RNA-Seq, metagenomica</a></em> </td>
    <td> <a href='http://www.bioinformatica.crs4.it/'>CRS4, Loc. Pixinamanna, Pula CA, Italy</a> </td>
    <td> <a href="mailto:paolo DOT uva AT crs4 DOT it">Paolo Uva</a> </td>
  </tr>
  <tr>
    <th> October 9-11 </th>
    <td> <em><a href='http://bioinfo.genotoul.fr/index.php?id=34&tx_seminars_pi1%5BshowUid%5D=43'>Galaxy Training Days</a></em> </td>
    <td> <a href='http://bioinfo.genotoul.fr/'>GenoToul bioinformatics facility</a>, INRA, Toulouse Auzeville, France </td>
    <td> <a href="mailto:sarah DOT maman AT toulouse AT inra AT fr">Sarah Maman</a> </td>
  </tr>
  <tr>
    <th> October 22-26 </th>
    <td> <em><a href='http://www.ashg.org/2013meeting/pages/workshops.shtml#high'>High Throughput Data Analysis and Visualization with Galaxy</a></em> </td>
    <td> <a href='http://www.ashg.org/2013meeting/'>ASHG 2013</a>, Boston, Massachusetts, United States </td>
    <td> <a href='/src/people/anton/index.md'>Anton Nekrutenko</a>, <a href='/src/people/jennifer-jackson/index.md'>Jennifer Jackson</a> </td>
  </tr>
</table>

<br />

# Galaxy Distributions

The most recent official distribution was on [June 3, 2013](/src/archive/dev-news-briefs/2013-06-03/index.md).  There was a minor [security patch released on June 12](https://lists.galaxyproject.org/archives/list/galaxy-announce@lists.galaxyproject.org/message/MELA4C4YJ32QUVFXLMTH3RXY4X2QLGOS/).

## CloudMan Release

<div class='right'><a href='/src/cloudman/index.md'><img src="/src/images/logos/CloudManWideBlackLogo.png" alt="CloudMan" width="300" /></a></div>

A new version of [CloudMan](/src/cloudman/index.md) was [released in July](/src/news/cloudman-release/index.md).

[CloudMan](/src/cloudman/index.md) offers an easy way to get a personal and completely functional instance of Galaxy in the cloud in just a few minutes, without any manual configuration.

**IMPORTANT - please read**

Any new cluster will automatically start using this version of CloudMan. Existing clusters will be given an option to do an automatic update once the main interface page is refreshed. Note that this upgrade is a major version upgrade and thus the migration is rather complicated. The migration process has been automated but will take a little while to complete. If you have made customizations to your cluster in terms of adding file systems, upgrading the database, or similar, we do not recommend you perform the upgrade. Note that this upgrade comes with (and requires) a new AMI (ami-118bfc78), which will automatically be used when starting an instance via [CloudLaunch](http://usegalaxy.org/cloudlaunch).

**This update brings a large number of updates and new features, the most prominent ones being:**
* Unification of `galaxyTools` and `galaxyData` file systems into a single `galaxy` filesystem. This change makes it possible to utilize the [Galaxy Tool Shed](/src/toolshed/index.md) when installing tools into Galaxy.
* Added initial support for Hadoop-type workloads
* Added initial support for cluster federation via HTCondor
* Added a new file system service for an instance's transient storage, allowing it to be used across the cluster over NFS
* Added a service for the Galaxy Reports webapp
* Added optional [Loggly](http://loggly.com/) based off-site logging support
* Added tags to all resources utilized by CloudMan

For more details on the new features, see the the [CHANGELOG](https://bitbucket.org/galaxy/cloudman/src/tip/CHANGELOG.md?at=default) and for even more details see, *all [291 commit messages](https://bitbucket.org/galaxy/cloudman/commits/all?search=35baec1%3A8bbae3f) from 7 contributors*.

<br />

# Tool Shed Contributions

<div class='right'><a href='http://toolshed.g2.bx.psu.edu/'><img src="/src/images/logos/ToolShed.jpg" alt="Galaxy Tool Shed" width=150 /></a></div>

* [varscan](http://bit.ly/12cDtT2): mutation callers for targeted, exome, & whole-genome resequencing; cancer genomics 
* [jointsnvmix](http://bit.ly/153MT1x): Somatic mutations caller using probabilistic graphical model w/ tumour/normal pairs 
* [scaffold](http://bit.ly/16gjK2h): Tool wrapper for the commercial proteomics application Scaffold 
* [trim_galore](http://bit.ly/16gk9BR): adaptive quality and adapter trimmer 
* [bismark](http://bit.ly/12kirxb): A flexible aligner and methylation caller for Bisulfite-Seq applications 
* [augustus](http://bit.ly/1ad06vO): eukaryotic gene predictor 
* [glimmer3](http://bit.ly/12OFpMD): microbial DNA gene finder. 
* [trna_prediction](http://bit.ly/1aPhAhh): Prediction of t-RNA with aragorn and tRNAscan-SE  
* [rseqc](http://bit.ly/130kkPv): converts RNA-seq data from .bam to .wig   
* [sickle](http://bit.ly/15gM7i4): A windowed adaptive trimming tool for FASTQ files using quality  
* [tabix](http://bit.ly/13nbawD): index TAB-delimited genome position file in.tab.bgz; creates an index file in.tab.bgz.tbi 
* [glimmer_gene_calling_workflow](http://bit.ly/1bhYbWm): Gene calling with glimmer and trainingset of known genes.  
* [vcftools](http://bit.ly/1bDK1xo): validate/merge/compare/calculate basic population genetic statistics on VCF. 
* [bcftools](http://bit.ly/146sigA): work with BCF data 
* [cmpfastq](http://bit.ly/14UNqnV): compare QC filtered fastq files  
* [fastQValidator](http://bit.ly/1dALG54): validates the format of fastq files 
* [bs_seeker2](http://bit.ly/18kf3eU): Bisulfite sequencing mapper
* [spades](http://bit.ly/1aLaW94): genome assembler for regular and single-cell projects. 

# Other News

* [Announcement for 2013 ISCB Wikipedia competition](http://bit.ly/15eEzxy); The [Galaxy page](http://bit.ly/gxywp) is eligible.
* Follow [Saket Choudhary's Google Summer of Code Galaxy project work on Blogspot](http://bit.ly/1abz1bv)
* [bioblend 0.3.3](http://bit.ly/1brfjsL): CloudMan and Galaxy API library documentation at ReadTheDocs.
