---
title: August 2018 Tool Shed contributions
---

[<img class="float-right" src="/src/images/galaxy-logos/galaxy-toolshed-300.png" alt="Galaxy ToolShed" width="180">](http://toolshed.g2.bx.psu.edu/)

Tools contributed to the Galaxy Project [ToolShed](http://toolshed.g2.bx.psu.edu/) in [August 2018](/src/news/2018-09-galaxy-update/index.md).

* [All monthly summaries](/src/toolshed/contributions/index.md)

## New Tools

* *From [jjjjia](https://toolshed.g2.bx.psu.edu/view/jjjjia):*
   * [cpo_prediction](https://toolshed.g2.bx.psu.edu/view/jjjjia/cpo_prediction):  parses cpo stuff. 
* *From [dereeper](https://toolshed.g2.bx.psu.edu/view/dereeper):*
   * [beagle](https://toolshed.g2.bx.psu.edu/view/dereeper/beagle):  Performs genotype calling, genotype phasing, imputation of ungenotyped markers, and identity-by-descent segment detection. 
   * [haplotype_analysis_workflow](https://toolshed.g2.bx.psu.edu/view/dereeper/haplotype_analysis_workflow):  Haplotype analysis workflow. 
   * [cytoscape](https://toolshed.g2.bx.psu.edu/view/dereeper/cytoscape):  Visualize interaction networks from JSON file. 
   * [sniplay3_gwas_workflow](https://toolshed.g2.bx.psu.edu/view/dereeper/sniplay3_gwas_workflow):  Tassel-based GWAS workflow (GLM model) including population structure and correction for structure. 
   * [haplophyle](https://toolshed.g2.bx.psu.edu/view/dereeper/haplophyle):  Create haplotype network from haplotype sequences. 
* *From [gga](https://toolshed.g2.bx.psu.edu/view/gga):*
   * [tripal_phylogeny_sync](https://toolshed.g2.bx.psu.edu/view/gga/tripal_phylogeny_sync):  Synchronize phylogeny data (from the Tripal tool suite). Galaxy tools allowing to load data into a remote Tripal server.    Tripal is a toolkit for construction of online biological (genetics, genomics, breeding, etc), community database,  and is a member of the GMOD family of tools. Tripal provides by default integration with the GMOD Chado database schema and Drupal, a popular Content Management Systems (CMS).    https://github.com/galaxy-genome-annotation/python-tripal.
* *From [cstrittmatter](https://toolshed.g2.bx.psu.edu/view/cstrittmatter):*
   * [skesa](https://toolshed.g2.bx.psu.edu/view/cstrittmatter/skesa):  skesa. SKESA -- Strategic Kmer Extension for Scrupulous Assemblies.
* *From [bgruening](https://toolshed.g2.bx.psu.edu/view/bgruening):*
   * [sklearn_build_pipeline](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_build_pipeline):  Wrapper for scikit learn tool: Pipeline Builder. Scikit-learn is an open source machine learning library written in python.  It provides different flavors of machine learning algorithms for classification,  regression, and clustering. It has designed to interoperate with python numerical  and scientific libraries Numpy and Scipy.    The official repository of Scikit-learn is at https://github.com/scikit-learn/scikit-learn.
   * [sklearn_searchcv](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_searchcv):  Wrapper for scikit learn tool: Hyperparameter Search. 
   * [sklearn_svm_classifier](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_svm_classifier):  Wrapper for scikit learn tool: Support vector machines (SVMs). 
   * [sklearn_nn_classifier](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_nn_classifier):  Wrapper for scikit learn tool: Nearest Neighbors Classification. 
* *From [greg](https://toolshed.g2.bx.psu.edu/view/greg):*
   * [plant_tribes_gene_family_scaffold_updater](https://toolshed.g2.bx.psu.edu/view/greg/plant_tribes_gene_family_scaffold_updater):  Contains a tool that adds a new genome to a scaffold installed into Galaxy by the PlantTribes Scaffolds Downloader data manager tool. Contains a tool that adds a new genome to a scaffold installed into Galaxy by the PlantTribes Scaffolds Downloader data manager tool.
   * [plant_tribes_gene_family_scaffold_loader](https://toolshed.g2.bx.psu.edu/view/greg/plant_tribes_gene_family_scaffold_loader):  Contains a tool that analyzes a PlantTribes scaffold and inserts information about it into the Galaxy PlantTribes database. Contains a tool that analyzes a PlantTribes scaffold and inserts information about it into the Galaxy PlantTribes database.
   * [extract_ipm_date_interval](https://toolshed.g2.bx.psu.edu/view/greg/extract_ipm_date_interval):  Contains a tool to extract a date interval from the data produced by the insect_phenology_model tool. Contains a tool that extracts a date interval from the (data) dataset collection produced by the insect_phenology_model tool and zooms into the interval by producing plots with daily ticks on the x axis instead of weekly) tick marks.  The (data) dataset collection consisting of the restricted date interval is also produced.
* *From [martasampaio](https://toolshed.g2.bx.psu.edu/view/martasampaio):*
   * [phage_promoters](https://toolshed.g2.bx.psu.edu/view/martasampaio/phage_promoters):  Get promoters of phage genomes. 
* *From [hepcat72](https://toolshed.g2.bx.psu.edu/view/hepcat72):*
   * [extractsplitreads_bwamem](https://toolshed.g2.bx.psu.edu/view/hepcat72/extractsplitreads_bwamem):  A wrapper for version 0.1.0 of the extractSplitReads_BwaMem script that comes with lumpy-sv version 0.2.14a. Get split-read alignments from bwa-mem in lumpy compatible format. Ignores reads marked as duplicates. Works on read or position sorted SAM input. Tested on bwa mem v0.7.5a-r405.
   * [svtyper](https://toolshed.g2.bx.psu.edu/view/hepcat72/svtyper):  Compute variant genotypes. Compute genotype of structural variants based on breakpoint depth.
* *From [nml](https://toolshed.g2.bx.psu.edu/view/nml):*
   * [biohansel](https://toolshed.g2.bx.psu.edu/view/nml/biohansel):  Heidelberg and Enteritidis SNP Elucidation. biohansel - Subtype microbial whole-genome sequencing (WGS) data using SNP targeting k-mer subtyping schemes.
* *From [galaxyp](https://toolshed.g2.bx.psu.edu/view/galaxyp):*
   * [maldi_quant_peak_detection](https://toolshed.g2.bx.psu.edu/view/galaxyp/maldi_quant_peak_detection):  Wrapper for the MALDIquant suite tool: MALDIquant peak detection. MALDIquant is free and open source software for the R environment Distinctive features include baseline subtraction methods such as TopHat or SNIP, peak alignment using warping functions, handling of replicated measurements as well as allowing spectra with different resolutions.
   * [maldi_quant_preprocessing](https://toolshed.g2.bx.psu.edu/view/galaxyp/maldi_quant_preprocessing):  Wrapper for the MALDIquant suite tool: MALDIquant preprocessing. 
* *From [mnhn65mo](https://toolshed.g2.bx.psu.edu/view/mnhn65mo):*
   * [get_worldclim_data](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/get_worldclim_data):  Tool to get bioclimatic data from worldclim. Tool to get bioclimatic data from worldclim. Use the raster package. return a netcdf or raster or gtif gis file. Also offer a map visualisation.
   * [regionalgam](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/regionalgam):  regionalgam tools. RegionalGAM tools inspired by the work of RetoSchmucki.
   * [spocc_occ](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/spocc_occ):  get sp occurence data. get sp occurence data using spocc r package. Data sources including : Global Biodiversity Information Facility ('GBIF'), 'USGSs' Biodiversity Information Serving Our Nation ('BISON'), 'iNaturalist', Berkeley 'Ecoinformatics' Engine, 'AntWeb', Integrated Digitized 'Biocollections' ('iDigBio'), 'VertNet', Ocean 'Biogeographic' Information System ('OBIS'), and Atlas of Living Australia ('ALA').
   * [butterfly_analysis](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/butterfly_analysis):  compare butterfly diversity and abundance across regions, observation networks, and production methods. compare butterfly diversity and abundance across regions, observation networks, and production methods. Rcode 3differents analysis.
   * [rdata_handler](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/rdata_handler):  tool to view and extract data from r binary file. tool to view and extract data from r binary file.
   * [gdal](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/gdal):  GDAL Geospatial Data Abstraction Library functions. Gdal is a translator library for raster and vector geospatial data formats that is released under an X/MIT style Open Source license by the Open Source Geospatial Foundation.
   * [netcdf_handler](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/netcdf_handler):  tools for handling netcdf data. 2 tools for handling netcdf data. The first give informations about the binary file's content, the second can extract these data.
   * [tabular_to_csv](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/tabular_to_csv):  transform tabular to csv. transform tabular to csv.
   * [vigiechiro](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/vigiechiro):  Vigiechiro tools. tools created by the vigiechiro team to analyses and identify chiro sounds files.
   * [csv_to_tabular](https://toolshed.g2.bx.psu.edu/view/mnhn65mo/csv_to_tabular):  transform csv to tabular. transform csv to tabular.
* *From [iuc](https://toolshed.g2.bx.psu.edu/view/iuc):*
   * [scpipe](https://toolshed.g2.bx.psu.edu/view/iuc/scpipe):  A flexible preprocessing pipeline for single-cell RNA-sequencing data. A preprocessing pipeline for single cell RNA-seq data that starts from the fastq files and produces a gene count matrix and a HTML report with associated quality control information. It can process fastq data generated by CEL-seq, MARS-seq, Drop-seq, Chromium 10x and SMART-seq protocols.
   * [heinz](https://toolshed.g2.bx.psu.edu/view/iuc/heinz):  An algorithm for identification of the optimal scoring subnetwork. An algorithm for identification of the optimal scoring subnetwork. This tool provides  a workflow over the output (a bunch of p-values) of differential expression analysis  (e.g., DESeq2) by fitting them into a distribution first, then transforming them into Heinz  scores. Based on the scores, Heinz is able to pinpoint an optimal subnetwork from the background  network made from such as KEGG and Reactome.
   * [jcvi_gff_stats](https://toolshed.g2.bx.psu.edu/view/iuc/jcvi_gff_stats):  Compute statistics from a genome annotation in GFF3 format (using JCVI Python utilities). 
   * [seurat](https://toolshed.g2.bx.psu.edu/view/iuc/seurat):  A toolkit for quality control, analysis, and exploration of single cell RNA sequencing data. Seurat aims to enable users to identify and interpret sources of heterogeneity from single cell transcriptomic measurements.
   * [exonerate](https://toolshed.g2.bx.psu.edu/view/iuc/exonerate):  Exonerate is a generic tool for pairwise sequence comparison. Exonerate is a generic tool for pairwise sequence comparison. It allows you to align sequences using a many alignment models, either exhaustive dynamic programming or a variety of heuristics.
* *From [ieguinoa](https://toolshed.g2.bx.psu.edu/view/ieguinoa):*
   * [data_manager_salmon_index_builder](https://toolshed.g2.bx.psu.edu/view/ieguinoa/data_manager_salmon_index_builder):  Generate indexes for Salmon. 


