---
title: Training Day
---
{{> Events/GCC2014/Header }}
<br />

{{> Events/GCC2014/LinkBox }}

<img class="float-left" src="/src/images/logos/GCC2014TDLogoSmall.png" alt="GCC2014 Training Day" />

The [2014 Galaxy Community Conference (GCC2014)](/src/events/gcc2014/index.md) started on June 30 with a *Training Day* featuring five parallel tracks, each with several two and a half hour workshops.  Tracks cover *using* Galaxy for biological research, and *deploying and managing* Galaxy instances, and everything in between.

Topics were [nominated](/src/events/gcc2014/training-day/index.md#nomination-voting-and-topic-selection) by the Galaxy Community in December and voted on in January.

Workshops will be hands-on and participants are strongly encouraged to bring a laptop and follow along.

Meeting participants select which Training Day topics to attend when they [register](/src/events/gcc2014/register/index.md).


## Day 0: Learning Galaxy

<table class="table">
  <tr class="th" >
    <th style=" text-align: center;"> Time </th>
    <th style=" text-align: center; width: 18%;"> Barber Room 302 </th>
    <th style=" text-align: center; width: 18%;"> Salon A Room 303 </th>
    <th style=" text-align: center; width: 19%;"> Salon B Room 303 </th>
    <th style=" text-align: center; width: 20%;"> Salon C Room 303 </th>
    <th style=" text-align: center; width: 19%;"> Multipurpose Room 324 </th>
  </tr>
  <tr>
    <th style=" text-align: right;"> 8:00 </th>
    <td colspan=5 style=" text-align: center;"> <strong>Registration Opens and Catered Breakfast</strong> </td>
  </tr>
  <tr>
    <th style=" text-align: right;"> 9:00 </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> <strong><a href='/src/events/gcc2014/training-day/index.md#visualization-of-ngs-data'>Visualization of NGS data</a></strong><div class='indent'><a href='/src/people/jeremy-goecks/index.md'>Jeremy Goecks</a> & <a href='/src/people/guerler/index.md'>Aysam Guerler</a></div> </td>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> <strong><a href='/src/events/gcc2014/training-day/index.md#raisins-and-rabbit-turds-ngs-quality-control-with-galaxy'>Raisins & Rabbit Turds: NGS Quality Control with Galaxy</a></strong><div class='indent'><a href='http://www.medicine.uiowa.edu/humangenetics/biostatistics/'>Tom Bair</a> & <a href='/src/people/jennifer-jackson/index.md'>Jennifer Jackson</a> </div> </td>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> <strong><a href='/src/events/gcc2014/training-day/index.md#galaxy-internals-flow-control-within-galaxy'>Galaxy Internals: Flow control within Galaxy</a></strong> <div class='indent'><a href='/src/people/james-taylor/index.md'>James Taylor</a></div> </td>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> <strong><a href='/src/events/gcc2014/training-day/index.md#galaxy-installation-and-administration'>Galaxy installation and administration</a></strong> <div class='indent'><a href='/src/people/nate/index.md'>Nate Coraor</a> & <a href='/src/people/john-chilton/index.md'>John Chilton</a> </div> </td>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> <strong><a href='/src/events/gcc2014/training-day/index.md#training-with-galaxy-a-genome-assembly-example'>Training with Galaxy: a Genome Assembly Example</a></strong><div class='indent'><a href='http://www.vlsci.org.au/researcher/sgladman'>Simon Gladman</a> & <a href='http://www.vlsci.org.au/researcher/alonie'>Andrew Lonie</a> </div> </td>
  </tr>
  <tr>
    <th style=" text-align: right;"> 11:30 </th>
    <td colspan=5 style=" text-align: center;"> <strong>Lunch</strong><br />Sponsored by <strong><a href='http://www.emc.com/isilon'>EMC Isilon</a></strong><br /><a href='http://www.emc.com/isilon'><img src="/src/images/logos/EMCIsilonLogo.jpg" alt="EMC Isilon" width="150" /></a> </td>
  </tr>
  <tr>
    <th style=" text-align: right;"> 12:30 </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> <strong><a href='/src/events/gcc2014/training-day/index.md#galaxy-on-a-cluster-user-and-project-management'>Galaxy on a Cluster - User and Project Management</a></strong> <div class='indent'> <a href='http://www.usit.uio.no/om/organisasjon/uav/itf/ft/ansatte/nikolaiv/index.html'>Nikolay Vazov</a> & <a href='http://www.usit.uio.no/om/organisasjon/uav/itf/ft/ansatte/katerim/index.html'>Katerina Michalickova</a></div> </td>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> <strong><a href='/src/events/gcc2014/training-day/index.md#galaxy-automation-using-the-api'>Galaxy Automation: Using the API</a></strong><div class='indent'><a href='/src/people/dannon-baker/index.md'>Dannon Baker</a> & <a href='/src/people/carl-eberhard/index.md'>Carl Eberhard</a></div> </td>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> <strong><a href='/src/events/gcc2014/training-day/index.md#tool-development-from-bright-idea-to-toolshed-designing-a-galaxy-tool'>Tool Development from bright idea to toolshed - Designing a Galaxy Tool</a></strong><div class='indent'> <a href='/src/people/greg_vonkuster/index.md'>Greg Von Kuster</a>, <a href='https://github.com/bgruening'>Björn Grüning</a> & <a href='http://www.hutton.ac.uk/staff/peter-cock'>Peter Cock</a> </div> </td>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> <strong><a href='/src/events/gcc2014/training-day/index.md#rna-seq-analysis-with-galaxy-and-the-tuxedo-suite'>RNA-Seq Analysis with Galaxy and the Tuxedo Suite</a></strong><div class='indent'> <a href='http://www.researchgate.net/profile/Saskia_Hiltemann/'>Saskia Hiltemann</a>, Youri Hoogstrate & <a href='http://wiki.data.dtls.nl/index.php/User:Leon_Mei'>Hailiang (Leon) Mei</a> </div> </td>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> <strong><a href='/src/events/gcc2014/training-day/index.md#3d-genome-analysis-with-galaxy'>3D Genome Analysis with Galaxy</a></strong><div class='indent'><a href='http://www.ous-research.no/home/hovig/Bioinformatics%20personnel/9807'>Jonas Paulsen</a>, <a href='http://www.mn.uio.no/math/english/people/aca/tonjegl/'>Tonje Lien Gulbrandsen</a>, <a href='http://www.ous-research.no/home/hovig/Bioinformatics%20personnel/4696'>Morten Johansen</a> & <a href='http://biolchem.bs.jhmi.edu/pages/facultydetail.aspx?FID=343'>Karen Reddy</a></div> </td>
  </tr>
  <tr>
    <th style=" text-align: right;"> 3:00 </th>
    <td colspan=5 style=" text-align: center;"> <strong>Break</strong> </td>
  </tr>
  <tr>
    <th style=" text-align: right;"> 3:30 </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> <strong><a href='/src/events/gcc2014/training-day/index.md#rna-seq-analysis-with-galaxy-and-alternative-tools'>RNA-Seq Analysis with Galaxy and Alternative Tools</a></strong> <div class='indent'> <a href='http://www.researchgate.net/profile/Saskia_Hiltemann/'>Saskia Hiltemann</a>, Youri Hoogstrate & <a href='http://wiki.data.dtls.nl/index.php/User:Leon_Mei'>Hailiang (Leon) Mei</a> </div> </td>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> <strong><a href='/src/events/gcc2014/training-day/index.md#tool-development-from-bright-idea-to-toolshed-data-managers'>Tool Development from bright idea to toolshed - Data Managers</a></strong><div class='indent'> <a href='https://www.msi.umn.edu/users/jj'>JJ Johnson</a> & <a href='/src/people/dan/index.md'>Dan Blankenberg</a> </div> </td>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> <strong><a href='/src/events/gcc2014/training-day/index.md#visualization-of-ngs-data'>Visualization of NGS data</a></strong><div class='indent'><a href='/src/people/jeremy-goecks/index.md'>Jeremy Goecks</a> and <a href='/src/people/guerler/index.md'>Aysam Guerler</a></div> </td>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> <strong><a href='/src/events/gcc2014/training-day/index.md#scriptable-bioinformatics-cloud-infrastructures-with-cloud-biolinux-cloudman-galaxy'>Scriptable Bioinformatics Cloud Infrastructures with Cloud BioLinux, CloudMan & Galaxy</a></strong> <div class='indent'><a href='http://www.jcvi.org/cms/about/bios/kkrampis/'>Ntino Krampis</a>, <a href='/src/people/enis-afgan/index.md'>Enis Afgan</a>, <a href='http://www.linkedin.com/in/rsanka'>Ravi Sanka</a>, <a href='http://bcbio.wordpress.com/about/'>Brad Chapman</a> </div> </td>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> <strong><a href='/src/events/gcc2014/training-day/index.md##galaxy-on-a-cluster-user-and-project-management'>Galaxy on a Cluster - User and Project Management</a></strong> <div class='indent'><a href='http://www.usit.uio.no/om/organisasjon/uav/itf/ft/ansatte/nikolaiv/index.html'>Nikolay Vazov</a> & <a href='http://www.usit.uio.no/om/organisasjon/uav/itf/ft/ansatte/katerim/index.html'>Katerina Michalickova</a></div> </td>
  </tr>
  <tr>
    <th style=" text-align: right;"> 6:00 </th>
    <td colspan=5 style=" text-align: center;"> <strong>Break</strong> </td>
  </tr>
  <tr>
    <th style=" text-align: right;"> 6:15 </th>
    <td colspan=5 style=" vertical-align: top; text-align: center;"> <strong>Dinner</strong> (on your own) / <strong><a href='/src/events/gcc2014/bofs/index.md'>Birds-of-a-Feather Flock 1</a></strong> </td>
  </tr>
  <tr>
    <th style=" text-align: right;"> 10:00 </th>
    <td colspan=5 style=" text-align: center;"> <strong>Finish</strong> </td>
  </tr>
</table>

<br />
<table class="table">
  <tr>
    <td style=" width: 5%; border: none;"> </td>
    <td style=" vertical-align: top; text-align: center; width: 20%; background-color: #F1DDCB;"> Biologist-centric </td>
    <td style=" width: 5%; border: none;"> </td>
    <td style=" vertical-align: top; text-align: center; width: 20%; background-color: #EFFFEF;"> Developer-centric </td>
  </tr>
</table>



# Prerequisites and Technology

Each topic lists prerequisites for what you should know, and what hardware and software you will need for each.

## Hardware

In *almost every* workshop participants will need a wifi-enabled laptop. 

## Software

In general:

 **All Workshops:**
   Require a web browser such as Chrome, Firefox, or Safari.  The latest version of Internet Explorer should also work.

 **[Deployment and Development Workshops](/src/index.md#deployment-and-development-workshops):**
   Require the [VirtualBox virtual machine (VM) software](/src/events/gcc2014/training-day/vms/index.md) on your laptop.  See the [Training Day VMs](/src/events/gcc2014/training-day/vms/index.md) page for more. This should be installed before you arrive at the conference.

See each individual workshop's list of prerequisite for specifics.

# Topics

## Using Galaxy Topics

### Raisins and Rabbit Turds: NGS Quality Control with Galaxy

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #F1DDCB;"> <a href='http://www.medicine.uiowa.edu/humangenetics/biostatistics/'>Tom Bair</a>, <a href='http://uiowa.edu/'>University of Iowa</a><br /><a href='/src/people/jennifer-jackson/index.md'>Jennifer Jackson</a>, <a href='http://www.psu.edu/'>Penn State University</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> Often the first step in next generation sequencing data analysis is <em>quality control</em>.  How reliable is the data?  Does it have GC bias, or inaccuracies at the read ends, or contamination, or barcode corruption, or any number of other conditions that need to be detected and dealt with before the science begins.  This workshop will provide hands-on experience performing quality control checks and how to get your data analysis-ready using Galaxy.<br /><br />This workshop is also a good introduction to Galaxy for those who are not familiar with it. <br /><br /><strong><a href='https://depot.galaxyproject.org/hub/attachments/documents/presentations/GCC2014_TD_RaisinsQAQC.pdf'>Slides</a></strong>, <strong><a href='http://usegalaxy.org/u/usinggalaxy4/p/raisins'>Page with Histories on UseGalaxy.org</a></strong> for continued learning</td>
    <td rowspan=2 style=" width: 16%;"> &nbsp;<a href='http://aws.amazon.com/'><img src="/src/images/logos/AWSLogo.png" alt="Amazon Web Services" width="120" /></a>&nbsp; <br />This workshop uses [AWS](http://aws.amazon.com/)-based compute infrastructure
 </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> * A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>


<span style="font-size: smaller;"> This title was inspired by [Richard Smith's talk on "Experimental design: the importance of filtering](https://wiki.uiowa.edu/display/galaxyusers/IIHG+Bioinformatics+Course)" at the [Iowa Iowa Institute for Human Genetics' Bioinformatics Short course](https://wiki.uiowa.edu/display/galaxyusers/IIHG+Bioinformatics+Course) </span>
 
### Training with Galaxy: a Genome Assembly Example

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #F1DDCB;"> <a href='http://www.vlsci.org.au/researcher/sgladman'>Simon Gladman</a>, <a href='http://www.vlsci.org.au/'>VLSCI</a> <br /><a href='http://www.vlsci.org.au/researcher/alonie'>Andrew Lonie</a>, <a href='http://unimelb.edu.au/'>University of Melbourne</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> The <a href='http://genome.edu.au'>Australian Genomics Virtual Laboratory (GVL)</a> has developed a range of online tutorials based on Galaxy to aid in training and dissemination of bioinformatics expertise.  The tutorials are completely self contained (data, workflows, rationale and background) and cover a range of introductory and advanced topics including genome assembly, variant detection and RNA-seq. This workshop will provide an overview of the available tutorials followed by a hands-on session based on a microbial genome assembly tutorial. To perform the analysis, participants will use cloud instances of the GVL platform.<br /><br /><strong><a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/LonieGVL.pdf'>Slides</a></strong> </td>
    <td rowspan=2 style=" width: 16%;"> &nbsp;<a href='https://genome.edu.au/wiki/GVL'><img src="/src/images/logos/gvl-300.png" alt="Genomics Virtual Lab" width="70" /></a>&nbsp; <br />This workshop uses [GVL](https://genome.edu.au/wiki/GVL)-based compute infrastructure
</td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> * A general knowledge of Galaxy (for example, you should be familiar with the material in [Galaxy 101](http://usegalaxy.org/galaxy101)), or attendance at the "[Raisins and Rabbit Turds: NGS Quality Control with Galaxy](/src/events/gcc2014/training-day/index.md#raisins-and-rabbit-turds-ngs-quality-control-with-galaxy)" session. 
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>


### RNA-Seq Analysis with Galaxy and the Tuxedo Suite

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #F1DDCB;"> <a href='http://www.researchgate.net/profile/Saskia_Hiltemann/'>Saskia Hiltemann</a>, <a href='http://www.erasmusmc.nl/'>Erasmus Medical Center</a><br />Youri Hoogstrate, <a href='http://www.erasmusmc.nl/'>Erasmus Medical Center</a><br /><a href='http://wiki.data.dtls.nl/index.php/User:Leon_Mei'>Hailiang (Leon) Mei</a>, <a href='https://www.lumc.nl/home/'>Leiden University Medical Center</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> This hands-on workshop will demonstrate basic RNA-Seq transcript level comparison analysis using the Tophat (Bowtie), Cufflinks, Cuffmerge and Cuffdiff tools in Galaxy. We will compare the expression of genes under two conditions.<br /><br />We will demonstrate this analysis both with an installed reference genome and with a non-installed organism.<br /><br />Sample datasets small enough to be successfully processed during the course of the seminar will be provided. Participants will perform the analyses themselves on the provided cloud instance of Galaxy. <br /><br /> <strong><a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/RNA-SeqTuxedoHadouts.pdf'>Handouts</a>, <a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/RNA-SeqTuxedoSlides.pdf'>Slides</a></strong> </td>
    <td style=" text-align: center; width: 16%;"> &nbsp;<a href='http://aws.amazon.com/'><img src="/src/images/logos/AWSLogo.png" alt="Amazon Web Services" width="120" /></a>&nbsp; <br />This workshop uses [AWS](http://aws.amazon.com/)-based compute infrastructure
 </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td colspan=2 style=" vertical-align: top; background-color: #F1DDCB;"> * A general knowledge of Galaxy and NGS quality control issues and tools, or attendance at the "[Raisins and Rabbit Turds: NGS Quality Control with Galaxy](/src/events/gcc2014/training-day/index.md#raisins-and-rabbit-turds-ngs-quality-control-with-galaxy)" session.
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>



### RNA-Seq Analysis with Galaxy and Alternative Tools

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #F1DDCB;"> <a href='http://www.researchgate.net/profile/Saskia_Hiltemann/'>Saskia Hiltemann</a>, <a href='http://www.erasmusmc.nl/'>Erasmus Medical Center</a><br />Youri Hoogstrate, <a href='http://www.erasmusmc.nl/'>Erasmus Medical Center</a><br /><a href='http://wiki.data.dtls.nl/index.php/User:Leon_Mei'>Hailiang (Leon) Mei</a>, <a href='https://www.lumc.nl/home/'>Leiden University Medical Center</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> The Tuxedo suite of RNA-Seq tools (Cuff*, Tophat, ...) are installed on many Galaxy instances, including <a href='/src/main/index.md'>Main </a> and <a href='/src/cloudman/index.md'>CloudMan</a> installs.  However, many other options are available.  For example, Htseq, EdgeR and DESeq are also widely used, take a different approach to RNA-Seq analysis and return different results from the Tuxedo suite. <br /><br />This workshop would introduce alternative methods for RNA-Seq analysis, cover how to install them from the <a href='/src/toolshed/index.md'>Tool Shed</a> and to test they are properly installed. The workshop could finish by comparing results from these tools with those from the Tuxedo suite. <br /><br /><strong><a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/RNA-SeqAltHandouts.pdf'>Handouts</a>, <a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/RNA-SeqAltSlides.pdf'>Slides</a>, <a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/RNA-SeqAltAnswers.pdf'>Handout Answers</a></strong></td>
    <td style=" text-align: center; width: 16%;"> &nbsp;<a href='http://aws.amazon.com/'><img src="/src/images/logos/AWSLogo.png" alt="Amazon Web Services" width="120" /></a>&nbsp; <br />This workshop uses [AWS](http://aws.amazon.com/)-based compute infrastructure
 </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td colspan=2 style=" vertical-align: top; background-color: #F1DDCB;"> * A general knowledge of Galaxy and NGS quality control issues and tools, or attendance at the "[Raisins and Rabbit Turds: NGS Quality Control with Galaxy](/src/events/gcc2014/training-day/index.md#raisins-and-rabbit-turds-ngs-quality-control-with-galaxy)" session.
 * Familiarity with the Tuxedo suite or attendance at the [RNA-Seq Analysis with Galaxy and the Tuxedo Suite](/src/events/gcc2014/training-day/index.md#rna-seq-analysis-with-galaxy-and-the-tuxedo-suite) session
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>


### Visualization of NGS data

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #F1DDCB;"> <a href='/src/people/jeremy-goecks/index.md'>Jeremy Goecks</a>, <a href='http://www.gwu.edu/'>George Washington University</a><br /><a href='/src/people/guerler/index.md'>Aysam Guerler</a>, <a href='http://jhu.edu/'>Johns Hopkins University</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> Different ways of visualizing NGS data more on downstream analysis such as heat maps, pathway networks and R based charts and graphs.  This workshop will cover both primary NGS analyses --alignments, variants, annotations -- as well as downstream options. <br /><br /> <strong><a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/GCC2014_Viz_Workshop.pdf'>Slides</a></strong> <br /><a href='https://dl.dropboxusercontent.com/u/4745528/galaxy-demos/create_trackster_viz.mov'>Video: Create Trackster (genome browser) visualization and explore data</a> <br /><a href='https://dl.dropboxusercontent.com/u/4745528/galaxy-demos/visual-analysis-variants.mov'>Video: Visual Analysis in Trackster and Sweepster</a> </td>
    <td rowspan=2 style=" text-align: center; width: 16%;"> &nbsp;<a href='http://aws.amazon.com/'><img src="/src/images/logos/AWSLogo.png" alt="Amazon Web Services" width="120" /></a>&nbsp; <br />This workshop uses [AWS](http://aws.amazon.com/)-based compute infrastructure
 </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> * A general knowledge of Galaxy (for example, you should be familiar with the material in [Galaxy 101](http://usegalaxy.org/galaxy101)), or attendance at the "[Raisins and Rabbit Turds: NGS Quality Control with Galaxy](/src/events/gcc2014/training-day/index.md#raisins-and-rabbit-turds-ngs-quality-control-with-galaxy)" session. 
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>


### 3D Genome Analysis with Galaxy

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> <a href='http://www.ous-research.no/home/hovig/Bioinformatics%20personnel/9807'>Jonas Paulsen</a>, <a href='http://www.uio.no'>University of Oslo</a><br /> <a href='http://www.mn.uio.no/math/english/people/aca/tonjegl/'>Tonje Lien Gulbrandsen</a>, <a href='http://www.uio.no'>University of Oslo</a><br /><a href='http://www.ous-research.no/home/hovig/Bioinformatics%20personnel/4696'>Morten Johansen</a>, <a href='http://www.uio.no'>University of Oslo</a><br /><a href='http://biolchem.bs.jhmi.edu/pages/facultydetail.aspx?FID=343'>Karen Reddy</a>, <a href='http://jhu.edu/'>Johns Hopkins University</a></td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> The session will introduce the basics of tracks and track types, and how these relate to hypothesis formulation and statistical analysis, using the Galaxy-based <a href='http://hyperbrowser.uio.no/'>Genomic HyperBrowser</a>. The emphasis will be on analysing, interpreting and integrating 3D genomic data (such as Hi-C), using the <a href='http://hyperbrowser.uio.no/3d'>HiBrowse system</a>. In addition to introducing the general concepts, the session will show examples on how 3D genome analyses can be combined with other <a href='http://hyperbrowser.uio.no/'>HyperBrowser</a> and Galaxy tools, in order to go from initial hypotheses to final results. <br /><br /><strong><a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/3DGenomeAnalysis.pdf'>Slides</a></strong> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td style=" vertical-align: top; background-color: #F1DDCB;"> * A general knowledge of Galaxy (for example, you should be familiar with the material in [Galaxy 101](http://usegalaxy.org/galaxy101)), or attendance at the "[Raisins and Rabbit Turds: NGS Quality Control with Galaxy](/src/events/gcc2014/training-day/index.md#raisins-and-rabbit-turds-ngs-quality-control-with-galaxy)" session. 
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>


## Deployment and Development Topics


### Galaxy Installation and Administration

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #EFFFEF;"> <a href='/src/people/nate/index.md'>Nate Coraor</a>, <a href='http://psu.edu/'>Penn State University</a><br /><a href='/src/people/john-chilton/index.md'>John Chilton</a>, <a href='http://psu.edu/'>Penn State University</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> Topics: <br /> &bull; Installing Galaxy on a standalone system<br /> &bull; Installing Galaxy in a cluster environment<br /> &bull; Common administrative tasks<div class='indent'> &bull; Tool installation (using Tool Shed and manually)<br /> &bull;  Reference genome installation and configuration<br /> &bull; Misc. (user authentication, data libraries, other...)</div> &bull; Upgrading<br />&bull; Troubleshooting <br /><br /><strong><a href='/src/events/gcc2014/training-day/admin-walkthrough/index.md'>Workshop Walkthrough</a></strong> </td>
    <td rowspan=2 style=" text-align: center; width: 16%;"> <a href='http://virtualbox.org'><img src="/src/images/logos/VirtualBox180.png" alt="Virtual Machine Images" width="80" /></a><br />This workshop will require that you have the [VirtualBox](http://virtualbox.org) player installed on your laptop.  </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> * Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.
* Secure Shell (SSH) client software such as [PuTTY](http://www.putty.org/) for Windows, or the [Terminal Application](http://guides.macrumors.com/Terminal) that comes with Mac OS.
* Virtual machine (VM) player software, [VirtualBox](http://virtualbox.org) is recommended and has been tested with the conference virtual machine images. 
* The [virtual machine image](/src/events/gcc2014/training-day/vms/index.md#2-download-needed-vms-on-your-laptop) for this workshop.
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>


### Galaxy on a Cluster - User and Project Management

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #EFFFEF;"> <a href='http://www.usit.uio.no/om/organisasjon/uav/itf/ft/ansatte/nikolaiv/index.html'>Nikolay Vazov</a>, <a href='http://www.usit.uio.no/'>University of Oslo</a><br /><a href='http://www.usit.uio.no/om/organisasjon/uav/itf/ft/ansatte/katerim/index.html'>Katerina Michalickova</a>, <a href='http://www.usit.uio.no/'>University of Oslo</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> Galaxy is more and more often used as a front-end to huge HPC resources. At the same time, the HPC facilities require solid user authentication procedures and accounting mechanisms allowing to control the use of HPC resources. We will provide an overview of issues and several possible approaches the problem. Participants will then install a specific third party solution (GOLD) into a test Galaxy. <br /><br /><strong><a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/UserProjectMgmt.pdf'>Slides</a></strong> </td>
    <td rowspan=2 style=" text-align: center; width: 16%;"> <a href='http://virtualbox.org'><img src="/src/images/logos/VirtualBox180.png" alt="Virtual Machine Images" width="80" /></a><br />This workshop will require that you have the [VirtualBox](http://virtualbox.org) player installed on your laptop. </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> * Experience maintaining a production Galaxy server (recommended)
* Secure Shell (SSH) client software such as [PuTTY](http://www.putty.org/) for Windows, or the [Terminal Application](http://guides.macrumors.com/Terminal) that comes with Mac OS.
* Virtual machine (VM) player software, [VirtualBox](http://virtualbox.org) is recommended and has been tested with the conference virtual machine images. 
* The [virtual machine image](/src/events/gcc2014/training-day/vms/index.md#2-download-needed-vms-on-your-laptop) for this workshop.
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>



### Galaxy Automation: Using the API

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #EFFFEF;"> <a href='/src/people/dannon-baker/index.md'>Dannon Baker</a>, <a href='http://www.jhu.edu/'>Johns Hopkins University</a> <br /><a href='/src/people/carl-eberhard/index.md'>Carl Eberhard</a>, <a href='http://www.jhu.edu/'>Johns Hopkins University</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> Galaxy has a growing API that allows for external programs to control the system, search the resources, and issue work requests. The session would cover programmatic access of the API either by direct REST web calls or by using the BioBlend/blend4j APIs. <br /><br /><strong><a href='/src/events/gcc2014/training-day/api/index.md'>Slides and Scripts</a></strong></td>
    <td rowspan=2 style=" text-align: center; width: 16%;"> <a href='http://virtualbox.org'><img src="/src/images/logos/VirtualBox180.png" alt="Virtual Machine Images" width="80" /></a><br />This workshop will require that you have the [VirtualBox](http://virtualbox.org) player installed on your laptop. </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> * Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.
* A knowledge of Python programming.
* Secure Shell (SSH) client software such as [PuTTY](http://www.putty.org/) for Windows, or the [Terminal Application](http://guides.macrumors.com/Terminal) that comes with Mac OS.
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>


### Tool Development from bright idea to toolshed - Designing a Galaxy Tool

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #EFFFEF;"> <a href='/src/people/greg_vonkuster/index.md'>Greg Von Kuster</a>, <a href='http://www.psu.edu/'>Penn State University</a><br /><a href='https://github.com/bgruening'>Björn Grüning</a>, <a href='http://www.uni-freiburg.de/'>University of Freiburg</a><br /><a href='http://www.hutton.ac.uk/staff/peter-cock'>Peter Cock</a>, <a href='http://www.hutton.ac.uk/'>James Hutton Institute</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> Galaxy provides an easy way to create reproducible, sharable, easy-to-use analytical workflows… if every step of the analysis has a galaxy tool available to perform that application.<br /><br />The Galaxy Toolshed offers a place to share tools that can be imported into a Galaxy Server to complete an analysis workflow.  Installation of a well-designed tool can be as simple as a couple button clicks by a Galaxy administrator.<br /><br />This session covers development process and the design considerations for stocking the toolshed with well-designed, easy-to-install tools.   We will design a couple tools, determining how to lay out the inputs and parameters, generate the command line with the cheetah template, and add test cases.  Then we’ll submit them to a toolshed, and install them in our galaxy server. <br /><br /><strong>Slides: <a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/DesigningATool.pdf'>PDF</a>, <a href='http://www.slideshare.net/pjacock/galaxy-tools'>SlideShare</a></strong> </td>
    <td style=" text-align: center; width: 16%;"> <a href='http://virtualbox.org'><img src="/src/images/logos/VirtualBox180.png" alt="Virtual Machine Images" width="80" /></a><br />This workshop will require that you have the [VirtualBox](http://virtualbox.org) player installed on your laptop. </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td colspan=2 style=" vertical-align: top; background-color: #EFFFEF;"> * Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.
* Secure Shell (SSH) client software such as [PuTTY](http://www.putty.org/) for Windows, or the [Terminal Application](http://guides.macrumors.com/Terminal) that comes with Mac OS.
* Virtual machine (VM) player software, [VirtualBox](http://virtualbox.org) is recommended and has been tested with the conference virtual machine images. 
* The [virtual machine image](/src/events/gcc2014/training-day/vms/index.md#2-download-needed-vms-on-your-laptop) for this workshop.
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>


### Tool Development from bright idea to toolshed - Data Managers

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #EFFFEF;"> <a href='https://www.msi.umn.edu/users/jj'>JJ Johnson</a>, <a href='http://umn.edu'>University of Minnesota</a><br /><a href='/src/people/dan/index.md'>Dan Blankenberg</a>, <a href='http://www.psu.edu/'>Penn State University</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> Galaxy tools can require installed reference data in order to be used effectively. For example, Bowtie requires prebuilt indexes in order to efficiently map sequences to a genome.<br /><br />Data Managers enable a Galaxy administrator to add reference data to a Galaxy server via the admin webpage.<br /><br />This session covers the tool and toolshed requirements for using reference data within galaxy tools, and the design and development of tool data managers to install reference data on a Galaxy server. <br /><br /><strong><a href='/src/events/gcc2014/training-day/data-managers/index.md'>Tutorial</a></strong> </td>
    <td rowspan=2 style=" text-align: center; width: 16%;"> <a href='http://virtualbox.org'><img src="/src/images/logos/VirtualBox180.png" alt="Virtual Machine Images" width="80" /></a><br />This workshop will require that you have the [VirtualBox](http://virtualbox.org) player installed on your laptop. </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> * Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.
* Secure Shell (SSH) client software such as [PuTTY](http://www.putty.org/) for Windows, or the [Terminal Application](http://guides.macrumors.com/Terminal) that comes with Mac OS.
* Virtual machine (VM) player software, [VirtualBox](http://virtualbox.org) is recommended and has been tested with the conference virtual machine images. 
* The [virtual machine image](/src/events/gcc2014/training-day/vms/index.md#2-download-needed-vms-on-your-laptop) for this workshop.
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>


### Galaxy Internals: Flow control within Galaxy

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> <a href='/src/people/james-taylor/index.md'>James Taylor</a>, <a href='http://www.jhu.edu/'>Johns Hopkins University</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> Galaxy deployers often face problems in customizing the galaxy instance because of the lack of documentation that talks about how the control flows within Galaxy when job is run. This workshop will help deployers understand the Galaxy's internals. <br /><br /><strong><a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/GalaxyInternals.pdf'>Slides</a>, <a href='http://www.slideshare.net/jxxxxtxxxxx/20143006-taylor-gccdan'>SlideShare</a></strong> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> * Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.
 </td>
  </tr>
</table>


### Scriptable Bioinformatics Cloud Infrastructures with Cloud BioLinux, CloudMan & Galaxy

<table class="table">
  <tr>
    <th style=" vertical-align: top;"> Instructors </th>
    <td colspan=2 style=" vertical-align: top; background-color: #EFFFEF;"> <a href='http://www.jcvi.org/cms/about/bios/kkrampis/'>Ntino Krampis</a>, <a href='http://www.jcvi.org/'>JCVI</a> <br /><a href='/src/people/enis-afgan/index.md'>Enis Afgan</a>, <a href='http://www.irb.hr/eng/'>Ruđer Bošković Institute (RBI)</a> <br /><a href='http://www.linkedin.com/in/rsanka'>Ravi Sanka</a>, <a href='http://www.jcvi.org/'>JCVI</a> <br /><a href='http://bcbio.wordpress.com/about/'>Brad Chapman</a>,  <a href='http://compbio.sph.harvard.edu/chb/'>Harvard University</a> </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Content </th>
    <td style=" vertical-align: top; background-color: #EFFFEF;"> This workshop will provide instruction on building bioinformatics infrastructures with Galaxy as front-end, combined with Cloud BioLinux for standardization and CloudMan for scalability in the back-end. It will be a technically-oriented workshop targeted to software developers, and will provide a tutorial how to jointly leverage the three systems for building bioinformatics applications on various cloud platforms including Amazon, OpenStack and Eucalyptus.<br /><br />The basics of deploying bioinformatics tools and pipelines on Galaxy running pre-configured on a Virtual Machine will be demonstrated. We will then move onto methods for standardizing deployment of complex bioinformatics pipelines through Galaxy by leveraging the Python Fabric scripts of Cloud BioLinux, in order to achieve interoperability and easy deployment across the various cloud platforms. The software blueprint of CloudMan for instantiating and using virtualized clusters connected to the Galaxy back-end will be presented, in addition to best practices for designing bioinformatics applications that leverage the distributed computing capabilities offered by the CloudMan framework.<br /><br />All concepts will be demonstrated through hands-on sessions where users will deploy tools through Galaxy, build VMs through Cloud BioLinux, instantiate clusters and data volumes and run distributed computing through CloudMan, using Amazon or Eucalyptus clouds. <br /><br /> <strong><a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/CloudMan.pdf'>CloudMan Slides</a>, <a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/CloudBioLinux.pdf'>CloudBioLinux Slides</a>, <a href='https://depot.galaxyproject.org/hub/attachments/events/gcc2014/training-day/ViralCloudResource.pdf'>Viral Cloud Slides</a></strong></td>
    <td style=" text-align: center; width: 16%;"> &nbsp;<a href='http://aws.amazon.com/'><img src="/src/images/logos/AWSLogo.png" alt="Amazon Web Services" width="120" /></a>&nbsp; <br />This workshop uses [AWS](http://aws.amazon.com/)-based compute infrastructure
 </td>
  </tr>
  <tr>
    <th style=" vertical-align: top;"> Prerequisites </th>
    <td colspan=2 style=" vertical-align: top; background-color: #EFFFEF;"> * Knowledge and comfort with the Unix/Linux command line interface and a text editor. If you don't know what cd, mv, rm, mkdir, chmod, grep and so on can do then you will struggle in this workshop.
* Secure Shell (SSH) client software such as [PuTTY](http://www.putty.org/) for Windows, or the [Terminal Application](http://guides.macrumors.com/Terminal) that comes with Mac OS.
* An account on [Amazon Web Services](https://aws.amazon.com).  These can be setup for free, but it does require a credit card.
* A wi-fi enabled laptop with a modern web browser.  Google Chrome, Firefox and Safari will work best. 
 </td>
  </tr>
</table>



# Nomination, Voting and Topic Selection

Training Day topics were selected by the Galaxy Community.  Topic were first nominated and then voted on by the community.  The schedule above is the direct result of that process.


# Feedback from the GCC2014 Training Day

<table class="table">
  <tr>
    <td style=" text-align: center; border: none;"> <span style="font-size: larger;"> "Have attended many training days/tutorials in my 15yr career. This was for me the most fruitful (and thus the best) so far." </span> </td>
  </tr>
  <tr>
    <td style=" text-align: center; border: none;"> "I could definitely tell a lot of time and energy went into planning this training day/conference, and for that I want to say thank you." </td>
  </tr>
  <tr>
    <td style=" text-align: center; border: none;"> "Intense but great" </td>
  </tr>
  <tr>
    <td style=" text-align: center; border: none;"> "All the instructors were amazing." </td>
  </tr>
  <tr>
    <td style=" text-align: center; border: none;"> "<strong>Overall a great experience</strong>" </td>
  </tr>
</table>


* <span style="font-size: larger;"> "Have attended many training days/tutorials in my 15yr career. This was for me the most fruitful (and thus the best) so far." </span>
* "I could definitely tell a lot of time and energy went into planning this training day/conference, and for that I want to say thank you." 
* "Intense but great" 
* "All the instructors were amazing."
* "**Overall a great experience**" 

# Training Day Sponsor

<div class='center'>
<a href='http://aws.amazon.com/'><img src="/src/images/logos/AWSLogo1280.jpg" alt="Amazon Web Services" width="250" /></a><br />

<br />

{{> Events/GCC2014/Footer }}
