---
title: February 2018 Tool Shed contributions
---

[<img class="float-right" src="/src/images/galaxy-logos/galaxy-toolshed-300.png" alt="Galaxy ToolShed" width="180">](http://toolshed.g2.bx.psu.edu/)

Tools contributed to the Galaxy Project [ToolShed](http://toolshed.g2.bx.psu.edu/) in [February 2018](/src/galaxy-updates/2018-03/index.md).

* [All monthly summaries](/src/toolshed/contributions/index.md)

## New Tools

### New Tools

* *From [proteore](https://toolshed.g2.bx.psu.edu/view/proteore):*
   * [proteore_expression_levels_by_tissue](https://toolshed.g2.bx.psu.edu/view/proteore/proteore_expression_levels_by_tissue):  ProteoRE Expression levels by tissue. A tool for selecting and/or annotating a list of ENSG IDs with tissue expression.
   * [proteore_id_converter](https://toolshed.g2.bx.psu.edu/view/proteore/proteore_id_converter):  ProteoRE ID_Converter: A tool converts identifiers which are of a different type/source to another type of identifiers and create the identifier lists. 
* *From [cole_easson](https://toolshed.g2.bx.psu.edu/view/cole_easson):*
   * [assembly](https://toolshed.g2.bx.psu.edu/view/cole_easson/assembly):  assembly. Velvet assembly tools.
* *From [rnateam](https://toolshed.g2.bx.psu.edu/view/rnateam):*
   * [peakachu](https://toolshed.g2.bx.psu.edu/view/rnateam/peakachu):  PEAKachu is a peak-caller for CLIP- and RIP-Seq data. PEAKachu is a peak-caller for CLIP- and RIP-Seq data. It takes input in BAM format and identifies regions of statistically significant read enrichment.  PEAKachu uses signal and control libraries (ideally more than three each) to detect binding sites. It implements two peak calling approaches.
* *From [bgruening](https://toolshed.g2.bx.psu.edu/view/bgruening):*
   * [sklearn_clf_metrics](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_clf_metrics):  Wrapper for scikit learn tool: Calculate metrics. Scikit-learn is an open source machine learning library written in python.  It provides different flavors of machine learning algorithms for classification,  regression, and clustering. It has designed to interoperate with python numerical  and scientific libraries Numpy and Scipy.    The official repository of Scikit-learn is at https://github.com/scikit-learn/scikit-learn.
   * [sklearn_ensemble](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_ensemble):  Wrapper for scikit learn tool: Ensemble methods. 
   * [svm_classifier](https://toolshed.g2.bx.psu.edu/view/bgruening/svm_classifier):  Wrapper for scikit learn tool: Support vector machines (SVMs). 
   * [sklearn_generalized_linear](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_generalized_linear):  Wrapper for scikit learn tool: Generalized linear models. 
   * [sklearn_sample_generator](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_sample_generator):  Wrapper for scikit learn tool: Generate. 
   * [sklearn_data_preprocess](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_data_preprocess):  Wrapper for scikit learn tool: Preprocess. 
   * [scipy_sparse](https://toolshed.g2.bx.psu.edu/view/bgruening/scipy_sparse):  Wrapper for scikit learn tool: Sparse Matrix Functions. 
   * [sklearn_discriminant_classifier](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_discriminant_classifier):  Wrapper for scikit learn tool: Discriminant Analysis. 
   * [nn_classifier](https://toolshed.g2.bx.psu.edu/view/bgruening/nn_classifier):  Wrapper for scikit learn tool: Nearest Neighbors Classification. 
   * [sklearn_pairwise_metrics](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_pairwise_metrics):  Wrapper for scikit learn tool: Evaluate pairwise distances. 
   * [sklearn_numeric_clustering](https://toolshed.g2.bx.psu.edu/view/bgruening/sklearn_numeric_clustering):  Wrapper for scikit learn tool: Numeric Clustering. 
   * [deeptools_estimatereadfiltering](https://toolshed.g2.bx.psu.edu/view/bgruening/deeptools_estimatereadfiltering):  Wrapper for the deepTools: estimateReadFiltering. deepTools address the challenge of visualizing the large amounts of data that  are now routinely generated from sequencing centers in a meaningful way. To do so,  deepTools contain useful routines to process the mapped reads data through removal of  duplicates and different filtering options to create coverage files in standard  bedGraph and bigWig file formats. deepTools allow the creation of normalized  coverage files or the comparison between two files (for example, treatment and control).    Finally, using such normalized and standardized files, multiple visualizations can be  created to identify enrichments with functional annotations of the genome.  For a gallery of images that can be produced and a description of the tools see http://f1000.com/posters/browse/summary/1094053    https://github.com/deeptools/deepTools  doi: 10.1093/nar/gku365  Wikipage: https://github.com/deeptools/deepTools/wiki    Repository-Maintainer: Björn Grüning    https://github.com/deeptools/deepTools.
   * [deeptools_alignmentsieve](https://toolshed.g2.bx.psu.edu/view/bgruening/deeptools_alignmentsieve):  Wrapper for the deepTools: alignmentsieve. 
* *From [caleb-easterly](https://toolshed.g2.bx.psu.edu/view/caleb-easterly):*
   * [peca](https://toolshed.g2.bx.psu.edu/view/caleb-easterly/peca):  peca. 
* *From [marie-tremblay-metatoul](https://toolshed.g2.bx.psu.edu/view/marie-tremblay-metatoul):*
   * [nmr_annotation](https://toolshed.g2.bx.psu.edu/view/marie-tremblay-metatoul/nmr_annotation):  [Metabolomics][W4M][NMR] NMR Annotation - Annotation and quantification of complex mixture NMR spectra. 
* *From [galaxyp](https://toolshed.g2.bx.psu.edu/view/galaxyp):*
   * [openms_percolatoradapter](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_percolatoradapter):  Wrapper for the OpenMS suite tool: PercolatorAdapter. OpenMS is an open-source software C++ library for LC/MS data management and analyses. It offers an infrastructure for the rapid development of mass spectrometry related software.  https://www.openms.de/.
   * [openms_openpepxllf](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_openpepxllf):  Wrapper for the OpenMS suite tool: OpenPepXLLF. 
   * [msi_filtering](https://toolshed.g2.bx.psu.edu/view/galaxyp/msi_filtering):  filtering of mass-spectrometry imaging data. This tool can filter mass-spectrometry imaging files for pixels and features of interest.
   * [openms_mapaligneridentification](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_mapaligneridentification):  Wrapper for the OpenMS suite tool: MapAlignerIdentification. 
   * [openms_cometadapter](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_cometadapter):  Wrapper for the OpenMS suite tool: CometAdapter. 
   * [openms_seedlistgenerator](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_seedlistgenerator):  Wrapper for the OpenMS suite tool: SeedListGenerator. 
   * [mass_spectrometry_imaging_segmentations](https://toolshed.g2.bx.psu.edu/view/galaxyp/mass_spectrometry_imaging_segmentations):  unsupervised spatial clustering. Performes unsupervised spatial clustering on mass spectrometry imaging data and provides images, plots and textfiles as outputs.
   * [openms_xfdr](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_xfdr):  Wrapper for the OpenMS suite tool: XFDR. 
   * [openms_psmfeatureextractor](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_psmfeatureextractor):  Wrapper for the OpenMS suite tool: PSMFeatureExtractor. 
   * [openms_speclibsearcher](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_speclibsearcher):  Wrapper for the OpenMS suite tool: SpecLibSearcher. 
   * [openms_mapalignerspectrum](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_mapalignerspectrum):  Wrapper for the OpenMS suite tool: MapAlignerSpectrum. 
   * [openms_metaboliteadductdecharger](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_metaboliteadductdecharger):  Wrapper for the OpenMS suite tool: MetaboliteAdductDecharger. 
   * [openms_openpepxl](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_openpepxl):  Wrapper for the OpenMS suite tool: OpenPepXL. 
   * [openms_siriusadapter](https://toolshed.g2.bx.psu.edu/view/galaxyp/openms_siriusadapter):  Wrapper for the OpenMS suite tool: SiriusAdapter. 
* *From [jfb](https://toolshed.g2.bx.psu.edu/view/jfb):*
   * [negative_motif_finder_7_7](https://toolshed.g2.bx.psu.edu/view/jfb/negative_motif_finder_7_7):  discover unphosphorylated substrates. This tool is intended for use in conjunction with KinaMine 7-7 and Kinatest 7-7.  This tool discovers all Y-centered motifs that could have been found by the KinaMine following the KALIP process, but weren't.  This script then infers that those motifs were NOT phosphorylated by the kinase, and outputs them.  That output is used in Kinatest 7-7.
   * [kinatest_r_7_7testing](https://toolshed.g2.bx.psu.edu/view/jfb/kinatest_r_7_7testing):  Performs KINATEST-ID algorithm. Requires: Positive Substrates, Negative Substrates, Substrate Background Frequency.  See this citation for details of its use: Lipchik, Andrew M., et al. "KINATEST-ID: a pipeline to develop phosphorylation-dependent terbium sensitizing kinase assays." Journal of the American Chemical Society 137.7 (2015): 2484-2494.
   * [kinamine7_7](https://toolshed.g2.bx.psu.edu/view/jfb/kinamine7_7):  Extract peptide motifs. This takes a Distinct Peptide Report and extracts from it all phospho-motifs that were discovered at a threshold above a given FDR value.  This tool is intended to be used in conjunction with Negative Motif Finder and Kinatest.R, the three together creating a GalaxyP version of the KINATEST-ID workbook.
* *From [sblanck](https://toolshed.g2.bx.psu.edu/view/sblanck):*
   * [smagexp_datatypes](https://toolshed.g2.bx.psu.edu/view/sblanck/smagexp_datatypes):  Datatypes for SMAGEXP tools. 
   * [smagexp](https://toolshed.g2.bx.psu.edu/view/sblanck/smagexp):  SMAGEXP (Statistical Meta Analalysis for Gene EXPression) for galaxy. SMAGEXP (Statistical Meta Analalysis for Gene EXPression) integrates metaMA and metaRNAseq packages into Galaxy. We aim to propose a unified way to carry out meta-analysis of gene expression data, while taking care of their specificities.  We have developed this tool suite to analyse microarray data from Gene Expression Omnibus (GEO) database or custom data from affymetrix microarrays. These data are then combined to carry out meta-analysis using metaMA package. SMAGEXP also offers to combine raw read counts from Next Generation Sequencing (NGS) experiments using DESeq2 and metaRNASeq package. In both cases, key values, independent from the technology type, are reported to judge the quality of the meta-analysis.
* *From [fabio](https://toolshed.g2.bx.psu.edu/view/fabio):*
   * [sbtas_se](https://toolshed.g2.bx.psu.edu/view/fabio/sbtas_se): 20180201. AllSome Sequence Bloom Tree Search Engine. Rapid querying of massive sequence datasets.
* *From [mingchen0919](https://toolshed.g2.bx.psu.edu/view/mingchen0919):*
   * [aurora_fastqc_site](https://toolshed.g2.bx.psu.edu/view/mingchen0919/aurora_fastqc_site):  Evaluate short reads with FastQC software on a single or a paired of untrimmed and trimmed reads files. 
   * [aurora_deseq2_site](https://toolshed.g2.bx.psu.edu/view/mingchen0919/aurora_deseq2_site):  Differential expression analysis with R DESeq2 package. 
* *From [vimalkumarvelayudhan](https://toolshed.g2.bx.psu.edu/view/vimalkumarvelayudhan):*
   * [viga](https://toolshed.g2.bx.psu.edu/view/vimalkumarvelayudhan/viga): Initial commit - v0.10.3 git commit deeded0. De novo VIral Genome Annotator. 
* *From [nml](https://toolshed.g2.bx.psu.edu/view/nml):*
   * [refseq_masher](https://toolshed.g2.bx.psu.edu/view/nml/refseq_masher):  Find what NCBI RefSeq genomes match or are contained within your sequence data using Mash_ with a Mash sketch database of 54,925 NCBI RefSeq Genomes. 
* *From [iuc](https://toolshed.g2.bx.psu.edu/view/iuc):*
   * [mageck_test](https://toolshed.g2.bx.psu.edu/view/iuc/mageck_test):  Wrapper for the MAGeCK tool suite: MAGeCKs test. Model-based Analysis of Genome-wide CRISPR-Cas9 Knockout (MAGeCK) is a computational tool to identify  important genes from the recent genome-scale CRISPR-Cas9 knockout screens technology.
   * [mageck_mle](https://toolshed.g2.bx.psu.edu/view/iuc/mageck_mle):  Wrapper for the MAGeCK tool suite: MAGeCK mle. 
   * [mageck_count](https://toolshed.g2.bx.psu.edu/view/iuc/mageck_count):  Wrapper for the MAGeCK tool suite: MAGeCK count. 
   * [mageck_gsea](https://toolshed.g2.bx.psu.edu/view/iuc/mageck_gsea):  Wrapper for the MAGeCK tool suite: MAGeCK GSEA. 
   * [mageck_pathway](https://toolshed.g2.bx.psu.edu/view/iuc/mageck_pathway):  Wrapper for the MAGeCK tool suite: MAGeCK pathway. 
   * [msaboot](https://toolshed.g2.bx.psu.edu/view/iuc/msaboot):  A multiple sequences alignment bootstrapping tool. This tool can be used to generate bootstrapped replicates of multiple sequence alignments. The bootstrapping is done on the columns of the alignment.
   * [umi_tools_whitelist](https://toolshed.g2.bx.psu.edu/view/iuc/umi_tools_whitelist):  Wrapper for the UMI-tools suite tool: UMI-tools whitelist. Extract UMI barcode from a read and add it to the read name, leaving  any sample barcode in place. Can deal with paired end reads and UMIs  split across the paired ends.
   * [fasttree](https://toolshed.g2.bx.psu.edu/view/iuc/fasttree):  FastTree infers approximately-maximum-likelihood phylogenetic trees from alignments of nucleotide or protein sequences - GVL. FastTree infers approximately-maximum-likelihood phylogenetic trees from alignments of nucleotide or protein sequences. This version is just suitable for Linux 64-bit.
   * [snippy](https://toolshed.g2.bx.psu.edu/view/iuc/snippy):  Contains the snippy tool for characterising microbial snps. Snippy finds variants between a haploid reference genome and your NGS sequence reads. It will find both substitutions and insertions/deletions (indels).
* *From [yating-l](https://toolshed.g2.bx.psu.edu/view/yating-l):*
   * [gonramp_apollo_tools](https://toolshed.g2.bx.psu.edu/view/yating-l/gonramp_apollo_tools):  A suite of tools for managing a (local) Apollo server. 
