---
title: August 2017 Galaxy News
---
<div class='right'>
[<img src="/src/images/galaxy-logos/GalaxyNews.png" alt="Galaxy News" width=150 />](/src/galaxy-updates/index.md)
</div>

Welcome to the August 2017 *[Galactic News](/src/galaxy-updates/index.md)*, a summary of what is going on in the Galaxy community.  If you have anything to add to next month's newsletter, then please send it to outreach@galaxyproject.org.


# GCC2018: Portland, Oregon, United States, June 25-30

<div class="left">
[<img src="/src/images/logos/gcc-bosc-2018-logo-300.png" width="200" alt="GCC2018 will be colocated with BOSC 2018 in Portland, Oregon">](https://gccbosc2018.sched.com/)
</div>

We are pleased to announce that the **2018 Galaxy Community Conference (GCC2018)** and the **[Bioinformatics Open Source Conference 2018 (BOSC2018)](https://www.open-bio.org/wiki/BOSC)** are [joining together for a unified event](https://gccbosc2018.sched.com/) to be held in Portland, Oregon, United States, June 25-30, 2018.  The event will feature two days of training, a two day meeting, and finish with two days of intense collaboration at the [OBF Codefest](https://www.open-bio.org/wiki/Codefest), and Hack the Galaxy: Dev and Data.  The meeting will feature joint and parallel sessions, and shared keynotes, poster and demo sessions, and social events.  Colocating GCC and BOSC will bring together the widest possible community of bioinformatics developers and practitioners into a single place.  At this event the whole will be greater than the sum of the parts.

The event is organized by [Oregon Health & Science University (OHSU)](https://www.ohsu.edu) and will be held at [Reed College](http://www.reed.edu/), a small liberal arts college in Portland.  Portland is a vibrant city in the American Northwest, close to the Cascade Mountains, the Oregon Coast, and the Columbia River. 


# GCC2017 is Done!

<div class="right">[<img src="/src/images/news-graphics/gcc2017-group-photo.jpg" alt='2017 Galaxy Community Conference (GCC2017)' width="160" />](https://gcc2017.sciencesconf.org/)
<br />
[Photo](https://www.flickr.com/photos/134305289@N03/34817510423/in/album-72157685584858176/): [Bérénice Batut](https://www.flickr.com/photos/134305289@N03/albums/72157685584858176)<br /><br />
[<img src="/src/images/logos/F1000research-logo.jpg" alt='GCC2017 writeup at F1000' width="160" />](https://blog.f1000.com/2017/07/31/intergalactique-the-galaxy-community-conference-2017/)
</div>

GCC2017 was held 26-30 June in [Montpellier, France](https://gcc2017.sciencesconf.org/page/location). The eighth global gathering of the Galaxy community featured two days of *[Hack the Galaxy](https://schd.ws/hosted_files/gcc2017/ea/S8_L8-9_gcc2017_hacks.pdf)*, [two days of training](https://gcc2017.sched.com/overview/type/D.x+Training+-+All), and a [two day community conference](https://gcc2017.sched.com/overview/type/E.x+Conference+-+All). Most training materials, slides, and posters are [now available online](https://gcc2017.sched.com/). Slides and posters are also being uploaded to the [F1000 Galaxy Channel](https://f1000research.com/gateways/galaxy). Take a look.

Two summaries of GCC2017 are available online:

* [Intergalactique – the Galaxy Community Conference 2017](https://blog.f1000.com/2017/07/31/intergalactique-the-galaxy-community-conference-2017/), by [Hollydawn Murray](https://blog.f1000.com/author/hmurray/), F1000 Research Publishing Editor, on the [F1000 Research Blog](https://blog.f1000.com/blogs/f1000research/).
* [Trip Blog](https://www.melbournebioinformatics.org.au/projects-blog/uk-tour-2017-blog2/), by [Simon Gladman](https://www.melbournebioinformatics.org.au/people/simon-gladman/), Research Scientist specialising in microbial genomics, [Melbourne Bioinformatics](https://www.melbournebioinformatics.org.au/).

<div class="left"><br />
[<img src="/src/images/logos/sacnas-no-text.jpg" alt="Preconference data workshops at 2017 SACNAS Meeting" width="150" />](/src/events/2017-sacnas/index.md) <br /><br />
[<img src="/src/images/logos/SoftwareCarpentryLogo.png" alt="Software Carpentry" width="150" />](https://software-carpentry.org/)<br /><br />
[<img src="/src/images/logos/DataCarpentryLogo.png" alt="Data Carpentry" width="150" />](https://data-carpentry.org/)<br /><br />
[<img src="/src/images/logos/cyverse_logo_trans.png" alt="CyVerse" width="150" />](https://www.cyverse.org/)<br /><br />
[<img src="/src/images/galaxy-logos/galaxy_project_logo_blue.png" alt="Galaxy" width="150" />](/src/index.md)
</div>

# Galaxy (and CyVerse and the Carpentries) at 2017 SACNAS Meeting

If you are attending the [2017 SACNAS meeting](http://www.2017sacnas.org/) in Salt Lake City in October, then please consider signing up for one of [these preconference workshops](/src/events/2017-sacnas/index.md):

**[Software and Data Carpentry](/src/events/2017-sacnas/index.md#workshop-1-software-and-data-carpentry)**

**[Data Platforms for Life Science Researchers](/src/events/2017-sacnas/index.md#workshop-2-data-platforms-for-life-science-researchers)**

[Registration](http://bit.ly/sacnas-data-reg-2017) is *free* but space is limited.  You are strongly encouraged to register soon.
Both workshops are held October 18, the day before SACNAS starts.

**[Register Now](http://bit.ly/sacnas-data-reg-2017)**


*Plese note:* The Galaxy Community Fund is augmenting [SACNAS Travel Scholarships awards](http://www.2017sacnas.org/events/2017-sacnas-the-national-diversity-in-stem-conference/custom-167-344a437cdada442d9179834c162b0a21.aspx) for the first ~ 28 scholarship recipients that register for these workshops. The additional funds will cover an extra night's lodging for scholarship recipients, enabling you to arrive the day before the workshops. Funds are limited, and once the funds are allocated, they are gone. (You must hve received a SACNAS Travel Scholarship to be eligible.)

# Galaxy @ ISMB/ECCB/BOSC 2017: Slides and posters are available

<div class="right">
[<img src="/src/images/logos/ismb-2017.png" alt="Galaxy at ISMB/ECCB/BOSC 2017Slides and posters" width="120" />](/src/events/2017-ismb-bosc/index.md))
</div>

If you missed all the [Galaxy @ ISMB/ECCB 2017 & BOSC 2017](/src/events/2017-ismb-bosc/index.md) then don't, um, fret!  Most slides and posters are [available online](/src/events/2017-ismb-bosc/index.md))

See you next year in Chicago!

# All events

There are [a plenitude of Galaxy related events coming up](/src/events/index.md) in the next few months:

<table class="table table-striped">
  <thead>
    <tr>
      <th>Date</th>
      <th>Topic/Event</th>
      <th>Venue/Location</th>
      <th>Contact</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><span class="text-nowrap">August 2nd 2017</span></td>
      <td><a href="http://qfab.org/workshops/genome-assembly-using-galaxy-2-3-aug-2017">Genome assembly using Galaxy</a></td>
      <td><img style="float:right" src="/images/icons/AU.png"/><a href="http://qfab.org/">QFAB, St Lucia, Queensland, Australia</a></td>
      <td><a href="https://training.galaxyproject.org/"><img style="float:right" alt="Training offered by GTN Member" src="/images/galaxy-logos/GTN16.png" title="Training offered by GTN Member"/></a>training@qfab.org</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">August 9th 2017</span></td>
      <td><a href="http://www.polyomics.gla.ac.uk/course-galaxy-workshop_Aug17.html">Introduction to Galaxy Workshop</a></td>
      <td><img style="float:right" src="/images/icons/EU.png"/><a href="http://www.polyomics.gla.ac.uk/index.html">University of Glasgow, Glasgow, United Kingdom</a></td>
      <td>David Meltzer</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">September 1st 2017</span></td>
      <td><a href="http://www.earlham.ac.uk/genome-10k-and-genome-science">The Galaxy Bioinformatics Platform: learning to use Galaxy for RNA-seq analysis</a> [<a href="/events/2017-genome-10k">hub page</a>]</td>
      <td><img style="float:right" src="/images/icons/EU.png"/><a href="http://www.earlham.ac.uk/">Genome 10K and Genome Science, Earlham Institute, Norwich, United Kingdom</a></td>
      <td><a href="https://training.galaxyproject.org/"><img style="float:right" alt="Training offered by GTN Member" src="/images/galaxy-logos/GTN16.png" title="Training offered by GTN Member"/></a>Nicola Soranzo, Graham Etherington</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">September 13th 2017</span></td>
      <td><a href="http://qfab.org/workshops/rna-seq-analysis-using-galaxy-13-14-sep-2017">RNA-Seq analysis using Galaxy</a></td>
      <td><img style="float:right" src="/images/icons/AU.png"/><a href="http://qfab.org/">QFAB, St Lucia, Queensland, Australia</a></td>
      <td><a href="https://training.galaxyproject.org/"><img style="float:right" alt="Training offered by GTN Member" src="/images/galaxy-logos/GTN16.png" title="Training offered by GTN Member"/></a>training@qfab.org</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">September 25th 2017</span></td>
      <td><a href="http://compgen.mdc-berlin.de/">Computational Genomics and RNA Biology</a></td>
      <td><img style="float:right" src="/images/icons/EU.png"/><a href="https://www.mdc-berlin.de/bimsb">Berlin Institute for Medical Systems Biology, Max Delbrück Center, Berlin, Germany</a></td>
      <td><a href="https://training.galaxyproject.org/"><img style="float:right" alt="Training offered by GTN Member" src="/images/galaxy-logos/GTN16.png" title="Training offered by GTN Member"/></a>compgen@mdc-berlin.de</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">October 3rd 2017</span></td>
      <td><a href="https://www.eventbrite.com.au/e/introduction-to-galaxy-the-genomics-virtual-laboratory-registration-36300240998">Introduction to Galaxy &amp; the Genomics Virtual Laboratory</a></td>
      <td><img style="float:right" src="/images/icons/AU.png"/><a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics, Carlton, Victoria, Australia</a></td>
      <td><a href="https://training.galaxyproject.org/"><img style="float:right" alt="Training offered by GTN Member" src="/images/galaxy-logos/GTN16.png" title="Training offered by GTN Member"/></a></td>
    </tr>
    <tr>
      <td><span class="text-nowrap">October 4th 2017</span></td>
      <td><a href="https://www.eventbrite.com.au/e/rna-seq-differential-gene-expression-analysis-using-galaxy-the-gvl-registration-36300315220">RNA-Seq Differential Gene Expression Analysis using Galaxy &amp; the GVL</a></td>
      <td><img style="float:right" src="/images/icons/AU.png"/><a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics, Carlton, Victoria, Australia</a></td>
      <td><a href="https://training.galaxyproject.org/"><img style="float:right" alt="Training offered by GTN Member" src="/images/galaxy-logos/GTN16.png" title="Training offered by GTN Member"/></a></td>
    </tr>
    <tr>
      <td><span class="text-nowrap">October 4th 2017</span></td>
      <td><a href="http://destair.bioinf.uni-leipzig.de/event/de-stair-1-workshop/">A primer for RNA-Seq processing, interpreting and visualization</a></td>
      <td><img style="float:right" src="/images/icons/EU.png"/><a href="https://www.google.com/maps/place/University+of+Freiburg,+Faculty+of+Biology/@48.0099215,7.8554122,17z/data=!3m1!4b1!4m5!3m4!1s0x47911c876bfa3fef:0x349b8880ff5e7f3b!8m2!3d48.0099215!4d7.8576009?hl=en">University of Freiburg, Freiburg Germany</a></td>
      <td>Steve Hoffmann, Olaf Wolkenhauer, Wolfgang Hess</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">October 5th 2017</span></td>
      <td><a href="https://www.eventbrite.com.au/e/variant-calling-with-galaxy-the-gvl-for-beginners-registration-36300355340">Variant Calling with Galaxy &amp; the GVL for beginners</a></td>
      <td><img style="float:right" src="/images/icons/AU.png"/><a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics, Carlton, Victoria, Australia</a></td>
      <td><a href="https://training.galaxyproject.org/"><img style="float:right" alt="Training offered by GTN Member" src="/images/galaxy-logos/GTN16.png" title="Training offered by GTN Member"/></a></td>
    </tr>
    <tr>
      <td><span class="text-nowrap">October 18th 2017</span></td>
      <td><a href="/events/2017-sacnas">SACNAS 2017 Pre-Conference Workshops</a></td>
      <td><img style="float:right" src="/images/icons/NA.png"/><a href="http://www.2017sacnas.org/">2017 SACNAS, Salt Lake City, Utah, United States</a></td>
      <td><a href="https://training.galaxyproject.org/"><img style="float:right" alt="Training offered by GTN Member" src="/images/galaxy-logos/GTN16.png" title="Training offered by GTN Member"/></a>Camille Avestruz, Tracy Teal, Joslynn Lee, Dave Clements</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">October 23rd 2017</span></td>
      <td><a href="http://sciencegateways.org/gateways2017/">Gateways 2017</a></td>
      <td><img style="float:right" src="/images/icons/NA.png"/><a href="https://www.umich.edu/">University of Michigan, Ann Arbor, Michigan, United States</a></td>
      <td>Enis Afgan</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">October 23rd 2017</span></td>
      <td><a href="http://www.denbi.de/symposium2017">The Future Development of Bioinformatics in Germany and Europe</a></td>
      <td><img style="float:right" src="/images/icons/EU.png"/><a href="http://www.uni-bielefeld.de/ZIF/">ZiF, Bielefeld University, Bielefeld, Germany</a></td>
      <td>Rolf Backofen, Björn Grüning</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">October 23rd 2017</span></td>
      <td><a href="http://cnrsformation.cnrs.fr/stage-17012-Analyse-avancee-de-sequences.html?axe=65">Analyse avancée de séquences</a></td>
      <td><img style="float:right" src="/images/icons/EU.png"/><a href="https://www.cbib.u-bordeaux.fr/">Carreire de l'Université Bordeaux Segalen, Bordeaux, France</a></td>
      <td>Aurélien Barré, Raluca Uricaru, Benjamin Dartigues</td>
    </tr>
    <tr>
      <td><span class="text-nowrap">January 8th 2018</span></td>
      <td><a href="https://www.elixir-europe.org/events/european-galaxy-administrator-workshop">European Galaxy Administrator Workshop</a></td>
      <td><img style="float:right" src="/images/icons/EU.png"/>Oslo, Norway</td>
      <td><a href="https://training.galaxyproject.org/"><img style="float:right" alt="Training offered by GTN Member" src="/images/galaxy-logos/GTN16.png" title="Training offered by GTN Member"/></a></td>
    </tr>
  </tbody>
</table>

See the [Galaxy Events Google Calendar](http://bit.ly/gxycal) for details on other events of interest to the community.


----

# New Publications

**298 new publications** referencing, using, extending, and implementing Galaxy were added to the [Galaxy CiteULike Group](http://www.citeulike.org/group/16008/) in June and July.

Some highlights from the recently added publications:

* [Create, run, share, publish, and reference your LC–MS, FIA–MS, GC–MS, and NMR data analysis workflows with the Workflow4Metabolomics 3.0 Galaxy online infrastructure for metabolomics](https://doi.org/10.1016/j.biocel.2017.07.002)
  Yann Guitton, Marie Tremblay-Franco, Gildas Le Corguillé, et al. *The International Journal of Biochemistry & Cell Biology* (July 2017), doi:10.1016/j.biocel.2017.07.002

* [Workflow Integration and Orchestration, Opportunities and the Challenges](https://doi.org/10.1007/978-3-319-56982-6_8)
  G. Kousalya, P. Balakrishnan, C. Pethuru Raj. In Automated Workflow Scheduling in Self-Adaptive Clouds (2017), pp. 137-156, doi:10.1007/978-3-319-56982-6_8

* [The RNA workbench: best practices for RNA and high-throughput sequencing bioinformatics in Galaxy](https://doi.org/10.1093/nar/gkx409)
  Björn A. Grüning, Jörg Fallmann, Dilmurat Yusuf, et al. *Nucleic Acids Research*, Vol. 45, No. W1. (03 July 2017), pp. W560-W566, doi:10.1093/nar/gkx409

* [RNA-bioinformatics: Tools, services and databases for the analysis of RNA-based regulation](https://doi.org/10.1016/j.jbiotec.2017.05.019)
  Rolf Backofen, Jan Engelhardt, Anika Erxleben, et al. *Journal of Biotechnology* (May 2017), doi:10.1016/j.jbiotec.2017.05.019

* [Assessing therapeutic potential of molecules: molecular property diagnostic suite for tuberculosis (MPDS<sup>TB</sup>)](https://doi.org/10.1007/s12039-017-1268-4)
  AnamikaSingh Gaur, Anshu Bhardwaj, Arun Sharma, et al. *Journal of Chemical Sciences*, Vol. 129, No. 5. (2017), pp. 515-531, doi:10.1007/s12039-017-1268-4

* [Computational Generation of RNA Nanorings](https://doi.org/10.1007/978-1-4939-7138-1_2)
  Rishabh Sharan, Eckart Bindewald, WojciechK Kasprzak, BruceA Shapiro; edited by Eckart Bindewald, Bruce A. Shapiro. In RNA Nanostructures, Vol. 1632 (2017), pp. 19-32, doi:10.1007/978-1-4939-7138-1_2

* [Globus: A Case Study in Software As a Service for Scientists](https://doi.org/10.1145/3086567.3086570)
  Bryce Allen, Rachana Ananthakrishnan, Kyle Chard, et al. In Proceedings of the 8th Workshop on Scientific Cloud Computing (2017), pp. 25-32, doi:10.1145/3086567.3086570

* [Demonstrating Distributed Workflow Computing with a Federating Wide-Area File System](https://doi.org/10.1145/3093338.3093389)
  Philip D. Blood, Anjana Kar, Jason Sommerfield, Beth L. Eicher, Richard Angeletti, J. Ray Scott. In Proceedings of the Practice and Experience in Advanced Research Computing 2017 on Sustainability, Success and Impact (2017), doi:10.1145/3093338.3093389

* [Customized workflow development and data modularization concepts for RNA-Sequencing and metatranscriptome experiments](https://doi.org/10.1016/j.jbiotec.2017.06.1203)
  Steffen C. Lott, Markus Wolfien, Konstantin Riege, et al. *Journal of Biotechnology* (July 2017), doi:10.1016/j.jbiotec.2017.06.1203

* [TAREAN: a computational tool for identification and characterization of satellite DNA from unassembled short reads](https://doi.org/10.1093/nar/gkx257)
  Petr Novák, Laura Ávila Robledillo, Andrea Koblížková, Iva Vrbová, Pavel Neumann, Jiří Macas. *Nucleic Acids Research*, Vol. 45, No. 12. (07 July 2017), pp. e111-e111, doi:10.1093/nar/gkx257

* [proFIA: A data preprocessing workflow for Flow Injection Analysis coupled to High-Resolution Mass Spectrometry](https://doi.org/10.1093/bioinformatics/btx458)
  Alexis Delabrière, Ulli M. Hohenester, Benoit Colsch, Christophe Junot, François Fenaille, Etienne A. Thévenot. *Bioinformatics* (2017), doi:10.1093/bioinformatics/btx458

## Publication Topics

| # | Tag | | # | Tag | | # | Tag | | # | Tag | 
| ---: | --- | --- | ---: | --- | --- | ---: | --- | --- | ---: | --- |
| 167 | [methods](http://www.citeulike.org/group/16008/tag/methods) | | 55 | [workbench](http://www.citeulike.org/group/16008/tag/workbench) | | 55 | [usepublic](http://www.citeulike.org/group/16008/tag/usepublic) | | 35 | [usemain](http://www.citeulike.org/group/16008/tag/usemain) | 
| 20 | [uselocal](http://www.citeulike.org/group/16008/tag/uselocal) | | 20 | [isgalaxy](http://www.citeulike.org/group/16008/tag/isgalaxy) | | 19 | [refpublic](http://www.citeulike.org/group/16008/tag/refpublic) | | 16 | [tools](http://www.citeulike.org/group/16008/tag/tools) | 
| 15 | [reproducibility](http://www.citeulike.org/group/16008/tag/reproducibility) | | 9 | [cloud](http://www.citeulike.org/group/16008/tag/cloud) | | 8 | [other](http://www.citeulike.org/group/16008/tag/other) | | 7 | [unknown](http://www.citeulike.org/group/16008/tag/unknown) | 
| 4 | [shared](http://www.citeulike.org/group/16008/tag/shared) | | 2 | [howto](http://www.citeulike.org/group/16008/tag/howto) | | 1 | [visualization](http://www.citeulike.org/group/16008/tag/visualization) | | 1 | [usecloud](http://www.citeulike.org/group/16008/tag/usecloud) | 

----

# Who's Hiring

<div class='right'>[<img src="/src/images/GalaxyIsExpandingCloud.png" alt="Please Help! Yes you!" width="220" />]</div>

The Galaxy is expanding! Please help it grow.

* [Ingénieur spécialisé en NGS](https://www.sfbi.fr/content/ing%C3%A9nieur-sp%C3%A9cialis%C3%A9-en-ngs), [BIOGECO](http://www6.bordeaux-aquitaine.inra.fr/biogeco), Bordeaux, France
* [Ingénieur web et bases de données pour les analyses métagénomiques des écosystèmes fromagers](https://www.sfbi.fr/content/ing%C3%A9nieur-web-et-bases-de-donn%C3%A9es-pour-les-analyses-m%C3%A9tag%C3%A9nomiques-des-%C3%A9cosyst%C3%A8mes-fromage-0), Micalis / MaIAGE, Jouy-en-Josas, France 
* [Développeur d’outils MS/MS pour le projet Workflow4Metabolomics sous Galaxy](https://www.sfbi.fr/content/d%C3%A9veloppeur-d%E2%80%99outils-msms-pour-le-projet-workflow4metabolomics-sous-galaxy), Station Biologique de Roscoff, France
* [Bioinformatics Research Associate](https://academicpositions.harvard.edu/postings/7556), Huttenhower Lab, Harvard School of Public Health, Boston, Massachusetts, United States
* [Scientific Research Programmer](http://goo.gl/cMZVCn), [Sethuraman Lab](http://arunsethuraman.weebly.com/), [California State University San Marcos](https://www.csusm.edu/). Develop model-based population genomics pipelines.

Got a Galaxy-related opening? Send it to outreach@galaxyproject.org and we'll put it in the [Galaxy News feed](/src/news/index.md) and include it in next month's [update](/src/galaxy-updates/index.md).

<br />


----

# Public Galaxy Server News

There are [over 90 publicly accessible Galaxy servers](/src/use/index.md) and [six semi-public Galaxy services](/src/use/index.md).  Here's what happened with them in May

## Galaxy Integrated Omics (GIO)

* A [two minute demo of the mzIdentML visualisation plugin](https://youtu.be/Txf_91oF52c) that runs in the [Galaxy Integrated Omics (GIO)](http://gio.sbcs.qmul.ac.uk/) is now available.


## Usegalaxy.org

* Jupyter Interactive Environments now run on Jetstream cloud via Docker Swarm and Galaxy Swarm Manager.


## New Public Galaxy Servers

## ABiMS

<div class='right solid'><a href='http://abims.sb-roscoff.fr'><img src="http://abims.sb-roscoff.fr/sites/abims.sb-roscoff.fr/files/ABiMS.png" alt="ABiMS" width="200" /></a></div>

[ABiMS Galaxy](http://galaxy3.sb-roscoff.fr/)  supports general purpose genomics analysis, featuring many standard tools plus many additional tools. However, we are specialized in RNASeq with reference and *de novo* RNASeq. Anyone with a professional/institutional email address can [request an account](http://abims.sb-roscoff.fr/account). The server is only available for registered users, and the default quota for new users is 20GB but can be extended on request.
 [Email support](mailto:support DOT abims AT sb-roscoff DOT fr) is available.  ABiMS is supported by [CNRS](http://www.cnrs.fr/), [UPMC](http://www.upmc.fr/), [BioGenOuest](https://www.biogenouest.org/), [IFB](http://www.france-bioinformatique.fr/), and [Station Biologique de Roscoff](http://www.sb-roscoff.fr/)

<!--

# Galaxy Community Directories

<table class="table">
  <tr>
    <td style=" text-align: center; border: none; width: 20%;"> <a href='https://training.galaxyproject.org/'><img src="/src/images/galaxy-logos/GTNLogo300.png" alt="Galaxy Training Network" height="100" /></a> </td>
    <td style=" text-align: center; border: none; width: 20%;"> <a href='/src/community/logs/index.md'><img src="/src/images/logos/LogBoardWText200.png" alt="Galaxy Community Log Board" height="100" /></a> </td>
    <td style=" text-align: center; border: none; width: 20%;"> <a href='/src/community/deployments/index.md'><img src="/src/images/logos/GalaxyDeploymentCatalog200.png" alt="Galaxy Deployment Catalog" height="100" /></a> </td>
  </tr>
  <tr>
    <td style=" text-align: center; vertical-align: top; border: none;"> <strong>Share your training resources and experience <em>now</em> </strong> </td>
    <td style=" text-align: center; vertical-align: top; border: none;"> <strong><a href='/src/community/logs/index.md#add-a-log-page'>Share your experience</a> <em><a href='/src/community/logs/index.md#add-a-log-page'>now</a></em> </strong> </td>
    <td style=" text-align: center; vertical-align: top; border: none;"> <strong><a href='/src/community/deployments/index.md#add-your-galaxy-deployment'>Describe your instance</a> <em><a href='/src/community/deployments/index.md#add-your-galaxy-deployment'>now</a></em> </strong> </td>
  </tr>
</table>

<br />

No new resources were added to community directories in January or February.
-->

----

# Tools

## ToolShed Contributions

<div class='right'>[<img src="/src/images/galaxy-logos/galaxy-toolshed-300.png" alt="Galaxy ToolShed" width="150"  />](http://toolshed.g2.bx.psu.edu/)</div>

[Tool Shed contributions from June and July](/src/toolshed/contributions/2017-07/index.md).

# Releases

## Galaxy TourBuilder

The Galaxy [TourBuilder](https://github.com/TailorDev/galaxy-tourbuilder) records your Galaxy analysis
in real-time and creates a Galaxy Tour for you - automatically! You can edit the tour simultanously, replay the
tour while you are analysing your data and much more!

Check out the [video](https://imgur.com/a/0YVvz) to get a taste of what it can do for you.

This extension is available under the name Galaxy Tour Builder in the [Chrome Web Store](https://chrome.google.com/webstore/detail/galaxy-tour-builder/mdfbapknmcpnbmggahhaegehbbbmhmgg), the Mozilla (Firefox) [Add-on website](https://addons.mozilla.org/de/firefox/addon/galaxy-tour-builder/), and the Opera [add-ons website](https://addons.opera.com/de/extensions/details/galaxy-tour-builder/).
Credits

This project was initiated during the [Swiss-Germany Galaxy Days](https://galaxyproject.org/events/sg2016/) a while ago in Freiburg by Dannon Baker and William Durand. During GCC 2017 Florian Eggenhofer, Matthias Bernt, Bérénice Batut and Clemens Blank have picked up this project and with tramendous help from [TailorDev](https://tailordev.fr) so a first version could be released.


## Planemo 0.41.0 - 0.44.0

<div class='right'>[<img src="/src/images/galaxy-logos/planemo-logo.png" alt="Planemo" width="200" />](https://pypi.python.org/pypi/planemo/)</div>

[Planemo](https://pypi.python.org/pypi/planemo) is a set of command-line utilities to assist in building tools for the Galaxy project.  These releases included [numerous fixes and enhancements](https://github.com/galaxyproject/planemo/blob/master/HISTORY.rst).

See [GitHub for details](https://github.com/galaxyproject/planemo/blob/master/HISTORY.rst).

## galaxy-lib 17.9.0-3

galaxy-lib is a subset of the Galaxy core code base designed to be used as a library. This subset has minimal dependencies and should be Python 3 compatible.  It's available from [GitHub](https://github.com/galaxyproject/galaxy-lib) and [PyPi](https://pypi.python.org/pypi/galaxy-lib).

This set of revisions includes:

* Bring in latest Galaxy dev changes.
* Implement support for building Singularity mulled containers.
* Implement mulled version 2 package hashing.
* Fix default namespace for mulled operations from mulled to biocontainers.

## gops 1.1.0

The [Galaxy Operations Python Package](https://github.com/galaxyproject/gops) now provides utilities for working with GFF files thanks to [@bgruening](https://github.com/bgruening) and [@nsoranzo](https://github.com/nsoranzo).


## Earlier Releases

Other packages that have been released in the prior 4 months.

### May 2017 Galaxy Release (v 17.05)

<div class="right">
<a href='http://getgalaxy.org'><img src="/src/images/logos/GalaxyNewLogo_GalaxyProject_Trans.png" alt="GalaxyProject" width="200" /></a><br /><br />
<a href="https://vimeo.com/216884518"><img src="/src/images/news-graphics/dataset-tagging-17-05.png" alt="Tag your datasets to keep track of where computed datasets came from" width="200" /></a><br /># Tags are propogated<br />in histories<br /><br />
<a href="https://depot.galaxyproject.org/hub/attachments/documents/videos/dataset-drag-17-05.gif"><img src="/src/images/news-graphics/dataset-drag-17-05.png" alt='Drag and drop with Galaxy Histories and Tools' width="200" /></a><br />Drag and drop in Galaxy 17.05
</div>

The [Galaxy Committers](https://github.com/galaxyproject/galaxy/blob/dev/doc/source/project/organization.rst) published the **[17.05 release of Galaxy](https://docs.galaxyproject.org/en/master/releases/17.05_announce.html)** in May.

**Tag your data with propagating hashtags**

* Large Galaxy histories used to be messy. Hashtags make it easy to track dataset (and collection) relationships.
* These two movies (both under a minute) explain how to use this [with datasets](https://vimeo.com/216884518) and [with collections](https://vimeo.com/216895965).
* Learn more about Galaxy histories in our updated [tutorial](https://galaxyproject.org/tutorials/histories/).

**Drag & Drop datasets into tool inputs**

* Interface now allows dragging datasets from history panel into the content selectors of the tool form.
* Implemented in [Pull Request 3871](https://github.com/galaxyproject/galaxy/pull/3871).

**Upload directly to a collection**

* You can now bypass the history manipulation and upload your data straight into a collection for convenience.
*  Learn more about collections and how to use them in a new [tutorial](https://galaxyproject.org/tutorials/collections/).

We extend special thanks to the 64 [New Contributors](https://docs.galaxyproject.org/en/master/releases/17.05_announce.html#new-contributors) to Galaxy in the past year.

Additionaly there are [New Configuration Options](https://docs.galaxyproject.org/en/master/releases/17.05_announce.html#new-configuration-options)  and [New Datatypes](https://docs.galaxyproject.org/en/master/releases/17.05_announce.html#new-datatypes) sections.


#### Galaxy Docker Image 17.05

The [Galaxy Docker project has seen a new release](https://github.com/bgruening/docker-galaxy-stable/releases/tag/17.05), following Galaxy 17.05. Major features are an additional Docker compose setup with SLURM and HT-Condor deployments, BioContainers integration and much more automatic testing.

And

* The [Galaxy Docker Project](https://github.com/bgruening/docker-galaxy-stable) has reached more than 23k downloads on [Dockerhub](https://hub.docker.com/r/bgruening/galaxy-stable/) - not counting quay.io and all flavors 
* From [John Chilton](/src/people/john-chilton/index.md): 
  * Awesome new mode for docker-galaxy-stable by  [Björn Grüning](/src/people/bjoern-gruening/index.md): [container scheduling with Condor](https://github.com/bgruening/docker-galaxy-stable/pull/323)


### Parsec 1.0.0 - 1.0.2

[Parsec](https://github.com/galaxy-iuc/parsec) is a newly released set of command-line utilities to assist in working with Galaxy servers.  It uses automatically generated wrappers for BioBlend functions. Manage histories, launch workflows, and more, all from the command line. The README includes several examples.



### BioBlend 0.9.0

[BioBlend](http://bioblend.readthedocs.org/) is a Python library for interacting with CloudMan and Galaxy‘s API.  BioBlend makes it possible to script and automate the process of cloud infrastructure provisioning and scaling via CloudMan, and running of analyses via Galaxy.

See the [release notes](https://github.com/galaxyproject/bioblend/releases) for what's new in release 0.9.0.


### Galaxy CloudMan 17.05 on AWS

A new release of [Galaxy CloudMan](/src/cloudman/index.md) is available on the Amazon Web Services cloud infrastructure. This release includes Galaxy 17.05, an updated [tool list](https://github.com/galaxyproject/galaxy-cloudman-playbook/blob/master/files/usegalaxy_main_tool_list.yaml), and Slurm configuration changes to improve job performance. To get started, vist https://beta.launch.usegalaxy.org/ or take a look at the [Getting Started](/src/cloudman/index.md#getting-started) guide.


### CloudLaunch ##

Technically, the [all-new Galaxy CloudLaunch service](https://beta.launch.usegalaxy.org/) has been in public beta since February but keep in mind that it will replace the [current CloudLaunch service](https://launch.usegalaxy.org/launch) eventually so give it a try and let us know how it performs for you.

### CloudBridge 0.2.0

[CloudBridge](https://github.com/gvlproject/cloudbridge) aims to provide a simple layer of abstraction over different cloud providers, reducing or eliminating the need to write conditional code for each cloud. It is currently under development and is in an Alpha state.  [Release 0.2.0](https://github.com/gvlproject/cloudbridge/releases/tag/v0.2.0) includes several fixes and enhancements.

### And the rest ...

Other Galaxy packages that haven't had a release in the past four months can be [found on GitHub](https://github.com/galaxyproject).

----


# Other News

<div class="right">[<img src="/src/images/news-graphics/tacc-mining-cancer-data-thumb.png" alt="Mining cancer data for treatment clues">](https://www.tacc.utexas.edu/-/mining-cancer-data-for-treatment-clues)</div>

* [Mining cancer data for treatment clues](https://www.tacc.utexas.edu/-/mining-cancer-data-for-treatment-clues): Cancer genomics researchers rely on TACC supercomputers to help identify risk factors and classify tumors
* [Five years of Galaxy-P!](https://twitter.com/usegalaxyp/status/888157211612110849)
* A [plugin framework for Galaxy tool error handling](https://github.com/galaxyproject/galaxy/pull/4305) (work-in-progress) by [Helena Rasche](https://github.com/hexylena). 
* [A Galaxy docker container with R markdown based Galaxy tools installed](https://github.com/statonlab/docker-GRReport).  This Docker image is used as a demonstration to show how to use R markdown as a unified framework for Galaxy tool development.
* From [John Chilton](/src/people/john-chilton/index.md): A cool outcome from our GCC hackathon - [support for Galaxy tool localization](https://github.com/galaxyproject/galaxy/pull/4251) (work in progress). 
* From Mallory Freeberg: [Differential abundance testing of small RNAs](https://galaxyproject.github.io/training-material/topics/transcriptomics/tutorials/srna/tutorial.html) tutorial
* From [Enis Afgan](/src/people/enis-afgan/index.md): [First launch of #usegalaxy on Microsoft #Azure!](https://twitter.com/enisafgan/status/880796649160880128) A huge thanks to RC Carter and Elizabeth Bruce for setting the wheels in motion.) A huge thanks to RC Carter and Elizabeth Bruce for setting the wheels in motion.

