---
title: July 2013 Galaxy Update
---
<div class='left'><a href='/src/galaxy-updates/index.md'><img src="/src/images/logos/GalaxyUpdate200.png" alt="Galaxy Updates" width=150 /></a></div>

Welcome to the July 2013 *[Galaxy Update](/src/galaxy-updates/index.md)*, a monthly summary of what is going on in the Galaxy community. *[Galaxy Updates](/src/galaxy-updates/index.md)* complement the *[Galaxy Development News Briefs](/src/docs/index.md)* which accompany new Galaxy releases and focus on Galaxy code updates.

<br />

# New Public Servers

Four new servers joined [the list of over 30 publicly accessible Galaxy servers](/src/use/index.md) in June.

<div class='left'><br /><div class='solid'><a href='http://p-galaxy.ddbj.nig.ac.jp/'><img src="/src/public-galaxy-servers/archive/ddbj-gil.jpg" alt="P-Galaxy" width="150" /></a></div></div>

## P-Galaxy

 
[P-Galaxy](http://p-galaxy.ddbj.nig.ac.jp/) is part of the [DDBJ Read Annotation Pipeline](http://dnaresearch.oxfordjournals.org/content/early/2013/05/07/dnares.dst017.full) and is described in "[DDBJ Read Annotation Pipeline: A Cloud Computing-Based Pipeline for High-Throughput Analysis of Next-Generation Sequencing Data](http://dnaresearch.oxfordjournals.org/content/early/2013/05/07/dnares.dst017.full)" by Nagasaki, *et al.* in *[DNA Research](http://dnaresearch.oxfordjournals.org/)*.  Support is available at [P-Galaxy Team](mailto:p-galaxy AT g DOT nig DOT ac DOT jp), and P-Galaxy is developed by the [Genome Informatics Laboratory](http://www.nig.ac.jp/section/yn/yn-e.html) and the [DNA Data Bank of Japan (DDBJ)](http://www.ddbj.nig.ac.jp/) of the [National Institute of Genetics (NIG)](http://www.nig.ac.jp/english/index.html).

## In Silico Galaxy

<div class='right solid'><a href='http://insilico.utulsa.edu/galaxy/'><img src="/src/images/logos/UTulsaLogo.png" alt="In Silico Galaxy" height="100" /></a></div>
From "[Encore: Genetic Association Interaction Network Centrality Pipeline and Application to SLE Exome Data](http://onlinelibrary.wiley.com/doi/10.1002/gepi.21739/abstract)," by Davis *et al.,* [In Silico Galaxy](http://insilico.utulsa.edu/galaxy/) implements [Encore](http://onlinelibrary.wiley.com/doi/10.1002/gepi.21739/abstract)
<div class='indent'>
an open source network analysis pipeline for genome-wide association studies and rare variant data. Encore constructs Genetic Association Interaction Networks or epistasis networks using two optional approaches: our previous information-theory method or a generalized linear model approach. Additionally, Encore includes multiple data filtering options, including Random Forest/Random Jungle for main effect enrichment and Evaporative Cooling and Relief-F filters for enrichment of interaction effects. Encore implements SNPrank network centrality for identifying susceptibility hubs (nodes containing a large amount of disease susceptibility information through the combination of multivariate main effects and multiple gene-gene interactions in the network)
</div>
[In Silico Galaxy](http://insilico.utulsa.edu/galaxy/) is developed and supported by the [In Silico Bioinformatics Lab](http://insilico.utulsa.edu/) at the [University of Tulsa](http://utulsa.edu/).

## LiSIs

<div class='left solid'><a href='http://lisis.cs.ucy.ac.cy/'><img src="/src/images/logos/LiSisLogo.png" alt="LiSIs" width="200px" /></a></div>

[LiSIs](http://lisis.cs.ucy.ac.cy/) is a platform for virtual screening.  The !LiSIs platform has been developed in the context of the cross-disciplinary [GRANATUM project](http://www.granatum.org) aiming to bridge the gap between biomedical researchers by ensuring their seamless access to the globally available information needed to perform complex experiments and to conduct studies on large-scale datasets. [LiSIs](http://lisis.cs.ucy.ac.cy/) is developed and maintained by the [e-Health Laboratory](http://www.medinfo.cs.ucy.ac.cy/) at the [Department of Computer Science, University of Cyprus](http://www.cs.ucy.ac.cy/).  [LiSIs](http://lisis.cs.ucy.ac.cy/) is an EU-funded project under FP7-(ICT-2009.5.3) and has its own [LiSIS Support Google Group](https://groups.google.com/forum/#!forum/lisis-support) for questions.
   
## Orione

<div class='right solid'><a href='http://orione.crs4.it/'><img src="/src/use/orione/orione-home-page-300.png" alt="Orione"  /></a></div>

[Orione](http://orione.crs4.it/) is a Galaxy based web server for microbiology. [Orione](http://orione.crs4.it/) includes all post mapping or assembling steps from scaffolding to complete annotation pipelines.  From the [GCC2013](/src/events/gcc2013/index.md) poster abstract "[Engaging Galaxy in Microbiology](/src/events/gcc2013/abstracts/posters/index.md#p7-engaging-galaxy-in-microbiology)"
 We started on selecting the relevant software in the microbiology area, developing then all the necessary tools to integrate them into the Galaxy ecosystem. In addition to that, we made available several specialized workflows covering major applications such as bacterial resequencing, de novo assembly, scaffolding, bacterial RNA-seq, gene annotation and metagenomics. Orione provides additional capabilities to perform integrative, reproducible and transparent bioinformatic data analysis in microbiology thus expanding the constellation of specialized Galaxy based web servers as Nebula, [Cistrome](/src/use/cistrome-analysis-pipeline/index.md) and several others.  

This installation of Galaxy has been configured such that anonymous users can operate in a limited way. If you need to store data on this website and/or use advanced Galaxy features such as sharing and workflows, please send us an [email](mailto:galaxyadmin AT crs4 DOT it) with a short request.


<div class='right'>
<table>
  <tr>
    <th> # </th>
    <th> Tag </th>
  </tr>
  <tr>
    <td style=" text-align: right;"> 22 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/methods'>methods</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 18 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/workbench'>workbench</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 8 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/usemain'>usemain</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 5 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/cloud'>cloud</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 4 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/unknown'>unknown</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 4 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/shared'>shared</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 3 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/refpublic'>refpublic</a>* </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 3 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/isgalaxy'>isgalaxy</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/uselocal'>uselocal</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/usepublic'>usepublic</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/tools'>tools</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/other'>other</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/project'>project</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/reproducibility'>reproducibility</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/visualization'>visualization</a>* </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/howto'>howto</a> </td>
  </tr>
  <tr>
    <td colspan=2 style=" text-align: center; border: none;"> <span style="font-size: smaller;"> * New tag this month </span> </td>
  </tr>
</table>


</div>


# New Papers

A record 53 new papers were added to the [Galaxy CiteULike Group](http://www.citeulike.org/group/16008/) in June. These papers may be particularly interesting to the Galaxy community

* "[Web-based visual analysis for high-throughput genomics](https://doi.org/10.1186/1471-2164-14-397)" by [Jeremy Goecks](/src/people/jeremy-goecks/index.md), [Carl Eberhard](/src/people/carl-eberhard/index.md), Tomithy Too, [Anton Nekrutenko](/src/people/anton/index.md), [James Taylor](/src/people/james-taylor/index.md), *BMC Genomics*, Vol. 14, No. 1. (2013), 397
* "[Detection of non-coding RNA in Bacteria and Archaea using the DETR’PROK Galaxy pipeline](https://doi.org/10.1016/j.ymeth.2013.06.003)" by Claire Toffano-Nioche, Yufei Luo, Claire Kuchly, Claire Wallon, Delphine Steinbach, Matthias Zytnicki, Annick Jacq, Daniel Gautheret, *[Methods](http://www.sciencedirect.com/science/journal/10462023)* (June 2013)
* "[Encore: Genetic Association Interaction Network Centrality Pipeline and Application to SLE Exome Data](https://doi.org/10.1002/gepi.21739)" by Nicholas A. Davis, Caleb A. Lareau, Bill C. White, *et al.* *Genetic Epidemiology* (June 2013)
* "[A review of Bioinformatics training applied to research in Molecular Medicine, Agriculture and Biodiversity in Costa Rica and Central America](https://doi.org/10.1093/bib/bbt033)" by Allan Orozco, Jessica Morera, Sergio Jiménez, Ricardo Boza, *Briefings in Bioinformatics* (30 May 2013)
* "[Streaming Support for Data Intensive Cloud-Based Sequence Analysis](https://doi.org/10.1155/2013/791051)" by Shadi A. Issa, Romeo Kienzler, Mohamed El-Kalioby, *et al.,* *BioMed Research International*, Vol. 2013 (2013), pp. 1-16
* "[Phase-defined complete sequencing of the HLA genes by next-generation sequencing](https://doi.org/10.1186/1471-2164-14-355)" by Kazuyoshi Hosomichi, Timothy Jinam, Shigeki Mitsunaga, Hirofumi Nakaoka, Ituro Inoue, *BMC Genomics*, Vol. 14, No. 1. (2013), 355
* "[Biology: The big challenges of big data](https://doi.org/10.1038/498255a)" by Vivien Marx, *Nature*, Vol. 498, No. 7453. (12 June 2013), pp. 255-260
* "[Software and supporting material for “SOAPdenovo2: An empirically improved memory-efficient short read de novo assembly](https://doi.org/10.5524/100044)" by Luo, *et al.* in [GigaDB](http://gigadb.org/dataset/100044)

The [1,000th Galaxy related paper](/src/news/galaxy-cite-u-like-paper1000/index.md) was added to the [Galaxy CiteULike Group](http://www.citeulike.org/group/16008/) in June.  Those first thousand papers are tagged as

<table>
  <tr>
    <th> # </th>
    <th style=" width: 20%;"> Tag </th>
    <td style=" border: none; width: 5%;"> </td>
    <th> # </th>
    <th style=" width: 20%;"> Tag </th>
    <td style=" border: none; width: 5%;"> </td>
    <th> # </th>
    <th style=" width: 20%;"> Tag </th>
  </tr>
  <tr>
    <td style=" text-align: right;"> 502 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/methods'>methods</a> </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 33 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/unknown'>unknown</a> </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 15 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/reproducibility'>reproducibility</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 325 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/workbench'>workbench</a> </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 29 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/shared'>shared</a> </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 9 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/uselocal'>uselocal</a>* </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 49 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/isgalaxy'>isgalaxy</a> </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 24 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/project'>project</a> </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 7 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/other'>other</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 46 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/tools'>tools</a> </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 23 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/howto'>howto</a> </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 6 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/usepublic'>usepublic</a>* </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 38 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/usemain'>usemain</a>* </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 18 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/cloud'>cloud</a>* </td>
    <td style=" border: none;"> </td>
    <td style=" text-align: right;"> 1 </td>
    <td> <a href='http://www.citeulike.org/group/16008/tag/usecloud'>usecloud</a>* </td>
  </tr>
  <tr>
    <td colspan=8 style=" text-align: center; border: none;"> </td>
  </tr>
  <tr>
    <td colspan=8 style=" text-align: center; border: none;"> <span style="font-size: smaller;"> * These tags were added in 2013 and very few papers before 2013 have been back-curated with them.  All the tags are explained on the <a href='/src/publication-library/index.md'>CiteULike</a> page. </span> </td>
  </tr>
</table>

<br />

# Who's Hiring

<div class='right'><a href='/src/galaxy-is-hiring/index.md'><img src="/src/galaxy-is-hiring/GalaxyIsHiringWordCloud2.png" alt="Please Help! Yes you!" width="200" /></a></div>

The Galaxy is expanding! Please help it grow.

* **The [Galaxy Project is hiring software engineers and post-docs](/src/galaxy-is-hiring/index.md)** at both Emory and Penn State.
* [Sr Bioinformatics Specialist, Tufts University](http://bit.ly/1bOLLBd), Boston MA.
* [Senior Developer](http://www.nature.com/naturejobs/science/jobs/321283-Senior-Developer-Stem-Cell-Bioinformatics-Core), Stem Cell Bioinformatics Core, Sage Bionetworks, Seattle, WA, United States
* [Bioinformatics Support Group Leader](http://bit.ly/VkRtrb) @ LSU

Got a Galaxy-related opening? Send it to outreach@galaxyproject.org and we'll put it in the [Galaxy News feed](/src/news/index.md) and include it in next month's [update](/src/galaxy-updates/index.md).

<br />

# Events

<div class='right'>
<a href='http://www.iscb.org/ismbeccb2013'><img src="/src/images/logos/ISMB2013Logo205.png" alt="ISMB/ECCB 2013" width="80" /></a>
</div>
<div class='right'>
<a href='http://www.open-bio.org/wiki/BOSC_2013'><img src="/src/images/logos/BOSC_logo.png" alt="BOSC 2013" width="60" /></a><br /><a href='http://igenomed2.stanford.edu/Proteomics2013_mssig/'><img src="/src/images/logos/MSSIG2013Logo.jpg" alt="MS SIG 2013: Beyond the Proteomics" width="60" /></a><br />
</div>

## ISMB / ECCB / BOSC / MS SIG 2013

[ISMB / ECCB 2013](http://www.iscb.org/ismbeccb2013) (along with http://www.open-bio.org/wiki/BOSC_2013|BOSC]], [MS SIG](http://igenomed2.stanford.edu/Proteomics2013_mssig/) and many other events) will be held in Berlin July 19-23.  As of this writing there are [10 talks, 9 posters, and 2 workshops](/src/events/ismb2013/index.md) related to Galaxy at the events.

## GCC2013

<div class='center'><a href='/src/events/gcc2013/index.md'><img src="/src/images/logos/GCC2013Logo400.png" alt="2013 Galaxy Community Conference (GCC2013)" width="300" /></a> </div>

<br />

The [2013 Galaxy Community Conference (GCC2013)](/src/events/gcc2013/index.md) starts *today* in Oslo Norway, at the [University of Oslo](http://uio.no).

<div class='right'><a href='/src/events/gcc2013/training-day/index.md'><img src="/src/images/logos/GCC2013TrainingDayLogo200.png" alt="2013 Galaxy Community Conference (GCC2013) Training Day" width="150" /></a></div>

[GCC2013](/src/events/gcc2013/index.md) is an opportunity to participate in [two full days](/src/events/gcc2013/program/index.md) of presentations, discussions, [poster sessions](/src/events/gcc2013/abstracts/posters/index.md), [lightning talks](/src/events/gcc2013/lightning/index.md) and [Birds of a Feather sessions](/src/events/gcc2013/bof/index.md), all about high-throughput biology and the tools that support it. The conference also includes a [Training Day](/src/events/gcc2013/training-day/index.md) for the second year in a row, this year with more in-depth topic coverage, more concurrent sessions, and more topics.  The conference ends with a closing dinner (sponsored by Ion Torrent) at the historic venue *Sporten* high above Oslo.  (And for the two evenings before that, we have reserved a pub...)

<br />

<div class='right'><br />{{> Events/GCC2013/BoF/LinkBox }}</div>

### Birds of a Feather Sessions

<div class='left'><a href='/src/events/gcc2013/bof/index.md'><img src="/src/images/logos/GCC2013BoFLogo.png" alt="" width="180" /></a></div>

Past Galaxy Community Conferences have been *the* event for networking in the Galaxy: There is no better place to meet and learn from others doing high-throughput biology.  GCC2013 extends this tradition by including *[Birds of a Feather (BoF)](/src/events/gcc2013/bof/index.md)* meetups at the event.  Birds of a Feather meetups are informal gatherings where participants group together based on common interests.  If you have something in the list at right you want to meet about, or you have a whole new topic, then please join or start a [GCC2013 BoF](/src/events/gcc2013/bof/index.md).

<br />

### Software Carpentry Boot Camp

Directly following GCC2013, there is a unique possibility to attend a [two-day Software Carpentry Boot Camp at the University of Oslo](http://bit.ly/14EmcSA) (in a building close to where the GCC is held). [Software Carpentry Boot Camps](http://software-carpentry.org/bootcamps/) aim to to help scientists and engineers become more productive by teaching them basic computing skills like program design, version control, testing, and task automation. In this two-day boot camp, short tutorials will alternate with hands-on practical exercises.

The course is aimed at postgraduate students and other scientists who are familiar with basic programming concepts (like loops, conditionals, arrays, and functions) but need help to translate this knowledge into practical tools to help them work more productively.

Content: The syllabus for this boot camp will include:

<div class='right'><a href='http://bit.ly/14EmcSA'><img src="/src/images/logos/SoftwareCarpentryLogo.png" alt="Software Carpentry Boot Camp, 3-4 July, University of Oslo" width="200" /></a></div>

* using the shell to do more in less time
* using version control to manage and share information
* basic Python programming
* how (and how much) to test programs

Visit the [Boot Camp Page](http://software-carpentry.org/bootcamps/2013-07-oslo.html) for more information, and registration.

<br />

## Other Upcoming Events

<div class='center'>
<a href='/src/events/gcc2013/index.md'><img src="/src/images/logos/GCC2013Logo200.png" alt="2013 Galaxy Community Conference (GCC2013)" width="135px" /></a>
<a href='http://www.qfab.org/events/upcoming/'><img src="/src/images/logos/QFABLogo.png" alt="QFAB Workshops" height="80" /></a>
<a href='/src/events/rt-workshop-tour2013/index.md'><img src="/src/images/logos/RTGalaxyWorkshopTour2013-300.png" alt="" width="140px" /></a>
<a href='/src/events/ismb2013/index.md'><img src="/src/images/logos/BOSC_logo.png" alt="BOSC 2013" width="100" /></a> &nbsp;
<a href='/src/events/ismb2013/index.md'><img src="/src/images/logos/ISMB2013Logo205.png" alt="ISMB/ECCB 2013" width="60" /></a> 
</div>

<div class='right'><a href='http://bit.ly/gxycal'><img src="/src/images/icons/CalendarIcon.gif" /></a></div> 
See the [Galaxy Events Google Calendar](http://bit.ly/gxycal) for details on these and other events.

<br /><br />

<table>
  <tr class="th" >
    <th> Date </th>
    <th> Topic/Event </th>
    <th> Venue/Location </th>
    <th> Contact </th>
  </tr>
  <tr>
    <th> June 30 - July&nbsp;2 </th>
    <td> <strong><em><a href='/src/events/gcc2013/index.md'>2013 Galaxy Community Conference (GCC2013)</a></strong></em> </td>
    <td> University of Oslo, Oslo, Norway </td>
    <td> <a href='/src/events/gcc2013/organizers/index.md'>Organizers</a> </td>
  </tr>
  <tr>
    <th> July 4 </th>
    <td> <em><a href='http://bit.ly/ZZe8Kt'>RNA-Seq analysis using Galaxy</a></em> </td>
    <td> <a href='http://www.qut.edu.au/about/campuses-and-locations/kelvin-grove'>QUT Kelvin Grove Campus</a>, Kelvin Grove, Australia </td>
    <td> <a href="mailto:m DOT crowe AT qfab DOT org">Mark Crowe</a> </td>
  </tr>
  <tr>
    <th rowspan=4> July 18-23 </th>
    <td> <em><a href='/src/events/niehs2013/index.md'>Introduction to Galaxy Workshop</a></em><div class='indent'>National Institute of Environmental Health Sciences (NIEHS)</div> </td>
    <td rowspan=4> <a href='/src/events/rt-workshop-tour2013/index.md'>2013 Research Triangle Workshop Tour</a>, North Carolina, United States </td>
    <td rowspan=4> <a href='/src/people/dave-clements/index.md'>Dave Clements</a> </td>
  </tr>
  <tr>
    <td> <em><a href='/src/events/unc2013/index.md'>Introduction to Galaxy Workshop</a></em> <div class='indent'>University of North Carolina, Chapel Hill</div> </td>
  </tr>
  <tr>
    <td> <em><a href='http://bit.ly/ZFVLG1'>Galaxy Installation Tutorial</a></em> <div class='indent'><strong><a href='http://bit.ly/ZFVLG1'>2013 GMOD Summer School</a></strong></div> </td>
  </tr>
  <tr>
    <td> <em><a href='/src/events/ncsu2013/index.md'>Introduction to Galaxy Workshop</a></em> <div class='indent'> North Carolina State University</div> </td>
  </tr>
  <tr>
    <th> July 19-23 </th>
    <td> <strong><a href='/src/events/ismb2013/index.md'>ISMB/ECCB, BOSC and MS SIG 2013</a></strong><div class='indent'>Talks, posters and workshops.  Lots of them.</div> </td>
    <td> Berlin, Germany </td>
    <td> <a href="mailto:outreach AT galaxyproject DOT org">Galaxy Outreach</a> </td>
  </tr>
  <tr>
    <th rowspan=3> July 21-25 </th>
    <td> <em><a href='http://bit.ly/116LWZB'>Experiences in building a Next-Generation Sequencing Analysis Service using Galaxy, Globus Online, and Amazon Web Services</a></em> </td>
    <td rowspan=3> <strong><a href='https://www.xsede.org/web/xsede13'>XSEDE13</a></strong>, San Diego, California, United States </td>
    <td> <a href='http://www.mcs.anl.gov/about/people_detail.php?id=347'>Ravi Madduri</a> </td>
  </tr>
  <tr>
    <td> <em><a href='http://bit.ly/Zx4rje'>A Sustainable National Gateway for Biological Computation</a></em> </td>
    <td> Josephine Palencia </td>
  </tr>
  <tr>
    <td> <em><a href='http://bit.ly/124Jm33'>Supporting Genomics and other Biological Research</a></em> </td>
    <td> Richard LeDuc </td>
  </tr>
  <tr>
    <th> September 28 - October&nbsp;1 </th>
    <td> <em>Galaxy Workshop</em> </td>
    <td> <a href='http://www.rcpa.edu.au/Continuing/CalendarOfEvents/September2013.htm'>The Genomic Bioinformatics Workshop</a>, Sydney, Australia </td>
    <td> <a href='/src/people/jennifer-jackson/index.md'>Jennifer Jackson</a>, <a href='/src/people/fubar/index.md'>Ross Lazarus</a> </td>
  </tr>
  <tr>
    <th> October 1-3 </th>
    <td> <a href='http://www.beyond-the-genome.com/'>Beyond the Genome 2013</a> </td>
    <td> San Francisco, California, United States </td>
    <td> <a href='/src/people/james-taylor/index.md'>James Taylor</a> </td>
  </tr>
  <tr>
    <th> October 22-26 </th>
    <td> <em><a href='http://www.ashg.org/2013meeting/pages/workshops.shtml#high'>High Throughput Data Analysis and Visualization with Galaxy</a></em> </td>
    <td> <a href='http://www.ashg.org/2013meeting/'>ASHG 2013</a>, Boston, Massachusetts, United States </td>
    <td> <a href='/src/people/james-taylor/index.md'>James Taylor</a>, <a href='/src/people/jennifer-jackson/index.md'>Jennifer Jackson</a> </td>
  </tr>
</table>


<br />

# Galaxy Distributions

<div class='right'>
<a href='/src/archive/dev-news-briefs/2013-06-03/index.md'><img src="/src/images/news-graphics/2013_06_03_scatterplot-1.png" alt="scatterplot" width=400 /></a></div>

The most recent official distribution was on [June 3, 2013](/src/archive/dev-news-briefs/2013-06-03/index.md).  There was minor  [security patch released on June 12](https://lists.galaxyproject.org/archives/list/galaxy-announce@lists.galaxyproject.org/thread/MELA4C4YJ32QUVFXLMTH3RXY4X2QLGOS/#MELA4C4YJ32QUVFXLMTH3RXY4X2QLGOS).

## June 3, 2013 Distribution

**Highlights:**

* Visualization tool updates to [Scatterplot](/src/archive/dev-news-briefs/2013-06-03/index.md#scatterplot) and [Trackster](/src/archive/dev-news-briefs/2013-06-03/index.md#trackster).
* Job distribution, error tracking/management, and reporting function improvements to [Admin](/src/archive/dev-news-briefs/2013-06-03/index.md#admin) & [Core](/src/archive/dev-news-briefs/2013-06-03/index.md#core). 

* Multiple [Tool](/src/archive/dev-news-briefs/2013-06-03/index.md#tools) updates, [History and Dataset](/src/archive/dev-news-briefs/2013-06-03/index.md#histories) upgrades, and other related [UI](/src/archive/dev-news-briefs/2013-06-03/index.md#ui) enhancements.
* New features and fixes added to the [Tool Shed](/src/archive/dev-news-briefs/2013-06-03/index.md#tool-shed) and related components. 
* [Python 2.5 Support officially ended](/src/archive/dev-news-briefs/2013-06-03/index.md#python_25_support_has_ended).
* Plus newly [merged Pull Requests](/src/archive/dev-news-briefs/2013-06-03/index.md#pull-requests-merged) and links to tickets covering key [Bug Fixes](/src/archive/dev-news-briefs/2013-06-03/index.md#bug-fixes).

*[http://getgalaxy.org](http://getgalaxy.org)*

*[http://bitbucket.org/galaxy/galaxy-dist](http://bitbucket.org/galaxy/galaxy-dist)*

*[http://galaxy-dist.readthedocs.org](http://galaxy-dist.readthedocs.org)*

```
new: $ hg clone https://bitbucket.org/galaxy/galaxy-dist#stable

upgrade: $ hg pull 
$ hg update release_2013.06.03
```


## June 12, 2013 Security Fix

A security vulnerability was recently discovered by Björn Grüning with Galaxy's "user impersonation" feature that can expose an administrator's active history to users whom they impersonate.  Only Galaxy instances with  
 ```
allow_user_impersonation = True
 ```

set in their configurations are affected, and only if an administrator makes use of the impersonation feature.  By default, user impersonation is disabled.

A fix (id: 9d42f1e32efb) has been provided in the stable branch of Galaxy.  To apply the fix, ensure you are on the stable branch and upgrade to the latest changeset:
 ```
% hg branch
stable

% hg pull -u
 ```

For Galaxy installations on relatively old versions that administrators are not yet ready to upgrade, there are three workarounds.  First, the patch can be downloaded and applied manually:
 ```
% wget -o security.patch https://bitbucket.org/galaxy/galaxy-central/commits/9d42f1e32efb654fda1e011dc66a4aa5888717f1/raw/
 ```

and then:
 ```
% hg patch security.patch
 ```

or:
 ```
% patch -p1 < security.patch
 ```

Second, the impersonation feature can be disabled by setting the following option in Galaxy's configuration file:
 ```
allow_user_impersonation = False
 ```

In all of the above cases, the Galaxy server process(es) must be restarted for the change to take effect.

Third, the feature can be left enabled and unpatched, and the vulnerability can be worked around by educating administrators who use the feature.  As long as a new history is created by the administrator prior to switching to the impersonated user, no data will be exposed to the impersonated user.

# Tool Shed Contributions

<div class='right'><a href='http://toolshed.g2.bx.psu.edu/'><img src="/src/images/logos/ToolShed.jpg" alt="Galaxy Tool Shed" width=150 /></a></div>

* [MMuFLR](http://bit.ly/19JoLFq): Missense Mutation and Frameshift Location Reporter Workflows. 
* [ensembl_vep](http://bit.ly/ZvuN8A): Annotation tool using Ensembl Variant Effect Predictor for GVL 
* [eqtl_tools](http://bit.ly/11OQqhR): eQTL Analysis using [PANAMA](http://bit.ly/120E35o)  
* [fasta_extract](http://bit.ly/10iT8yU): Extract a single sequence or set of sequences from a multifasta file 
* [fasta_stats](http://bit.ly/14vngXt): Produce basic statistics on a fasta/multifasta file  
* [repeat_masker](http://bit.ly/10iTw0k): RepeatMasker  
* [interproscan](http://bit.ly/12hEJ6B): Interproscan functional predictions of ORFs
* [fastq_validator](http://bit.ly/14i19Vn): Validate your FastQ Files. 
* [peptide_to_gff](http://bit.ly/13ht2iy): Outputs GFF3 with the peptide position in a reference genome 
* [cmpfastq](http://bit.ly/13htNrL): Compare two fastq files 
* [samifier](http://bit.ly/17u7xtT): integrate genomic & transcriptomic NGS data with proteomic MS data.
* [somatic_sniper](http://bit.ly/13eUa2H): identify single nucleotide positions differences between tumor and normal 

# Other News

* Tin-Lap Lee's slides from his Bio-IT 13 talk: [Next-Gen Sequencing Analysis by GigaGalaxy](http://www.slideshare.net/GigaScience/giga-galaxy-bioit-world-13tl)
* Now in MyExperiment: [Detrprok_wf](http://bit.ly/1aWNeVE): Detect 3 types of ncRNA. Uses s_mart & detrprok_scripts from toolshed 
* [Building a CloudMan instance from scratch](https://github.com/chapmanb/cloudbiolinux/blob/master/deploy/cloudman.md)
* [Tutorial: uploading NGS data to Amazon S3 for use in Galaxy](http://postolog.com/bbdPCT)
* [Video: ChIP-seq analysis using Galaxy](http://bit.ly/18S1eV5) by Ian Donaldson of U Manchester 
* At the University of Florida? UF Health is [offering 2 Galaxy Workshops, Basics on June 26 and NGS on July 2](http://bit.ly/15cV0Yj)
* [pycrac](http://bit.ly/11lEbNa): User-friendly Python tools for the analyses of CLIP/CRAC datasets 
* In Sweden? The Swedish National Infrastructure for Computing (SNIC) is [now supporting Galaxy (see last page)](http://bit.ly/11w8Tjf).
* Follow [Saket Choudhary's Google Summer of Code Galaxy project work on Blogspot](http://bit.ly/1abz1bv)
