---
title: The June 2015 Galactic News!
---
<div class='right'>
<table>
  <tr>
    <td style=" border: none; width: 28em;">  </td>
  </tr>
</table>

</div>

<div class='left'><a href='/src/galaxy-updates/index.md'><img src="/src/images/logos/GalaxyUpdate200.png" alt="Galaxy Updates" width=150 /></a></div>

Welcome to the June 2015 *[Galactic News](/src/galaxy-updates/index.md)*, a summary of what is going on in the Galaxy community. *[These newsletters](/src/galaxy-updates/index.md)* complement the *[Galaxy Development News Briefs](/src/docs/index.md)* which accompany new Galaxy releases and focus on Galaxy code updates.

# New Papers

70 new papers referencing, using, extending, and implementing Galaxy were added to the [Galaxy CiteULike Group](http://www.citeulike.org/group/16008/) in May. Highlights include:

* [BioMAJ2Galaxy: automatic update of reference data in Galaxy using BioMAJ](https://doi.org/10.1186/s13742-015-0063-8)
    by Anthony Bretaudeau, Cyril Monjeaud, Yvan Le Bras, Fabrice Legeai, Olivier Collin, *GigaScience*, Vol. 4, No. 1. (9 May 2015), doi:10.1186/s13742-015-0063-8
* [Design and bioinformatics analysis of genome-wide CLIP experiments](https://doi.org/10.1093/nar/gkv439), 
    by Tao Wang, Guanghua Xiao, Yongjun Chu, Michael Q. Zhang, David R. Corey, Yang Xie, *Nucleic Acids Research* (09 May 2015), gkv439, doi:10.1093/nar/gkv439
* [The Globus Galaxies platform: delivering science gateways as a service](https://doi.org/10.1002/cpe.3486)
    Ravi Madduri, Kyle Chard, Ryan Chard, Lukasz Lacinski, Alex Rodriguez, Dinanath Sulakhe, David Kelly, Utpal Dave and Ian Foster, *Concurrency and Computation: Practice and Experience* (April 2015), doi:10.1002/cpe.3486

The new papers were tagged with:

<table>
  <tr>
    <th> # </th>
    <th style=" width: 19%;"> Tag </th>
    <td rowspan=5 style=" border: none;"> &nbsp;&nbsp; </td>
    <th> # </th>
    <th style=" width: 19%;"> Tag </th>
    <td rowspan=5 style=" border: none;"> &nbsp;&nbsp; </td>
    <th> # </th>
    <th style=" width: 19%;"> Tag </th>
    <td rowspan=4 style=" border: none;"> &nbsp;&nbsp; </td>
    <th> # </th>
    <th style=" width: 19%;"> Tag </th>
  </tr>
  <tr>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://bit.ly/gxyculcloud'>Cloud</a>       </td>
    <td style=" text-align: right;"> - </td>
    <td> <a href='http://bit.ly/gxyculproject'>Project</a>                  </td>
    <td style=" text-align: right;"> 6 </td>
    <td> <a href='http://bit.ly/gxycultools'>Tools</a>       </td>
    <td style=" text-align: right;"> 10 </td>
    <td> <a href='http://bit.ly/gxyculusepublic'>UsePublic</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> - </td>
    <td> <a href='http://bit.ly/gxyculhowto'>HowTo</a>       </td>
    <td style=" text-align: right;"> 3 </td>
    <td> <a href='http://bit.ly/gxyculrefpublic'>RefPublic</a>              </td>
    <td style=" text-align: right;"> - </td>
    <td> <a href='http://bit.ly/gxyculusecloud'>UseCloud</a> </td>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://bit.ly/gxyculviz'>Visualization</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 3 </td>
    <td> <a href='http://bit.ly/gxyculisgalaxy'>IsGalaxy</a> </td>
    <td style=" text-align: right;"> 2 </td>
    <td> <a href='http://bit.ly/gxyculrepro'>Reproducibility</a>            </td>
    <td style=" text-align: right;"> 5 </td>
    <td> <a href='http://bit.ly/gxyculuselocal'>UseLocal</a> </td>
    <td style=" text-align: right;"> 14 </td>
    <td> <a href='http://bit.ly/gxyculworkbench'>Workbench</a> </td>
  </tr>
  <tr>
    <td style=" text-align: right;"> 38 </td>
    <td> <a href='http://bit.ly/gxyculmethods'>Methods</a>   </td>
    <td style=" text-align: right;"> 3 </td>
    <td> <a href='http://bit.ly/gxyculshared'>Shared</a>                    </td>
    <td style=" text-align: right;"> 10 </td>
    <td> <a href='http://bit.ly/gxyculusemain'>UseMain</a>   </td>
  </tr>
</table>


# Events

## GCC2015: 4-8 July, Norwich UK

<div class='right'><a href='http://gcc2015.tsl.ac.uk/'><img src="/src/images/logos/GCC2015LogoWide600.png" alt="GCC2015" width="170" /></a><br />
<table>
  <tr>
    <th> &nbsp;&nbsp; <a href='http://gcc2015.tsl.ac.uk/registration/'>Register Now</a> &nbsp;&nbsp; </th>
  </tr>
</table>

</div>

The **[2015 Galaxy Community Conference (GCC2015)](http://gcc2015.tsl.ac.uk/)** is the Galaxy community's annual gathering of users, developers, and administrators.  [Previous GCC's](/src/events/gcc2014/index.md) have drawn over 200 participants, and we expect that to happen again in 2015.  GCC2015 is being hosted by [The Sainsbury Lab](http://tsl.ac.uk/) in Norwich, UK, immediately before [BOSC](http://www.open-bio.org/wiki/BOSC_2015) and [ISMB/ECCB](http://www.iscb.org/ismbeccb2015) in Dublin.

There are a lot of events going on at GCC2015, including:

* **[Code Hackathon](http://gcc2015.tsl.ac.uk/Hackathon/)**, 4-5 July
* **[Data Wrangling Hackathon](http://gcc2015.tsl.ac.uk/data-hackathon/)**, 4-5 July (*new*)
* **[Training SunDay](http://gcc2015.tsl.ac.uk/Training-Day/)**, 5 July (*new and sold out*)
* **[Training Day](http://gcc2015.tsl.ac.uk/Training-Day/)**, 6 July
* **[GCC Meeting](http://gcc2015.tsl.ac.uk/Conference/)**, 7-8 July

### Registration is Open!

[registration](http://gcc2015.tsl.ac.uk/abstracts/registration/) is open, and there is still available space for most GCC2015 events. If you work in data-intensive life science research, then there is no meeting that will be more relevant this year than GCC2015.  We look forward to seeing you there.

**[Register now](http://gcc2015.tsl.ac.uk/registration/).**

### Late Poster Abstracts

There are still a limited number of poster slots available and we welcome [poster abstract submissions](http://gcc2015.tsl.ac.uk/Abstracts).  Poster authors will be notified of acceptance status within two weeks. *Please consider presenting your work. If you are dealing with big biological data, then this meeting wants to hear about it.*

You can also submit [late oral presentation submissions](http://gcc2015.tsl.ac.uk/Abstracts).  These will be considered as cancellations happen (and they usually do).


### Other Events Near GCC2015

Looking to extend your trip?  There are also **[at least 7 other events](/src/news/near-gcc2015/index.md)** going on in the British Isles in the week before and week after GCC2015, including events

<div class='right'><a href='/src/news/near-gcc2015/index.md'><img src="/src/news/near-gcc2015/MatrixSnap.png" alt="Events nearby GCC2015" width="150" /></a></div>

 In Norwich 
* [iPlant Tools & Services Workshop](http://www.tgac.ac.uk/361_Division/training-programme/courses-workshops/tgac-events/iplant-tools-and-services/)
* [BioJS Conference and Hackathon](http://conf.biojs.net/)

 In Cambridge
* [Exploring Human Host-Microbiome Interactions in Health and Disease](https://registration.hinxton.wellcome.ac.uk/display_info.asp?id=480)
* [Exploring Plant Variation Data](http://www.ebi.ac.uk/training/course/mining-plant-variation-data)
* [Human Genome Analysis: Genetic Analysis of Multifactorial Diseases](http://www.wellcome.ac.uk/Education-resources/Courses-and-conferences/Advanced-Courses-and-Scientific-Conferences/Advanced-Courses/WTX026851.htm)

 In Dublin
* [OpenBio Codefest 2015](http://www.open-bio.org/wiki/Codefest_2015)
* [ISMB / EECB 2015](http://www.iscb.org/ismbeccb2015) including [BOSC 2015](http://www.open-bio.org/wiki/BOSC_2015) and [BioVis 2015](http://www.biovis.net/) and [lots of Galaxy content](/src/events/ismb2015/index.md).

See the [Events Aplenty Near GCC2015](/src/news/near-gcc2015/index.md) news item for dates and deadlines.

<div class='right'>
<br />
<a href='https://www.coursera.org/specialization/genomics/41'><img src="/src/images/logos/GenomicDataScience.png" alt="Genomic Data Science Course" width="400" /></a><br />
<a href='https://www.coursera.org/specialization/jhudatascience/1'><img src="/src/images/logos/DataScienceSpecialization.png" alt="Data Science Specialization" width="400" /></a>
</div>

### GCC2015 Sponsorships

#### Hackathons Sponsor: Johns Hopkins University Data Science Specialization Program

We are pleased to announce that the [Johns Hopkins University Data Science Specialization Program](https://www.coursera.org/specialization/jhudatascience/1) offered on [Coursera](https://www.coursera.org/) is the exclusive sponsor for both the **GCC2015 [Coding Hackathon](http://gcc2015.tsl.ac.uk/Hackathon/) and the [Data Wrangling Hackathon](http://gcc2015.tsl.ac.uk/data-hackathon)**.

This program includes a [Genomic Data Science course](https://www.coursera.org/specialization/genomics/41) covering

* Basic computational, biological, and statistical skills for analyzing big genomic data
* Foundational tools for successfully engaging in this rapidly changing field
* How to go from unprocessed next generation sequencing data to meaningful biological results

and introduces Galaxy, Bioconductor, Python, command line, and [many other tools](https://www.coursera.org/specialization/genomics/41).

The series of nine MOOCs are now open for enrollment and free to anyone.

#### Training Day Sponsor: Amazon Web Services

<div class='left'>
<a href='http://aws.amazon.com/'><img src="/src/images/logos/AWSLogo1280.jpg" alt="Amazon Web Services" width="200" /></a>
</div>

We are pleased to again have [Amazon Web Services (AWS)](http://aws.amazon.com/) as the sponsor for the [GCC2015 Training Days](http://gcc2015.tsl.ac.uk/training-day/).  This is the 4th year in a row AWS has sponsored GCC Training Days.  Many of the sessions will be using AWS directly, or servers running on AWS.


#### Call for Sponsors

The [2015 Galaxy Community Conference (GCC2015)](http://gcc2015.tsl.ac.uk/) is still accepting [Sponsorships](http://gcc2015.tsl.ac.uk/sponsors/).  Your organisation can play a prominent part in the Galaxy community by sponsoring GCC2015.  Sponsorship is an excellent way to raise your organization’s visibility.

Several sponsorship levels are available, including two levels of premier sponsorships that include presentations.  Premium sponsorships are limited, however, so you are encouraged to act soon.

Please let the [organisers](http://gcc2015.tsl.ac.uk/) know if you are interested in helping make this event a success.

## June GalaxyAdmins Meetup

<div class='left'><a href='/src/community/galaxy-admins/meetups/2015-06-18/index.md'><img src="/src/images/logos/GalaxyAdmins.png" alt="GalaxyAdmins meetup June 18" height="75" /></a></div>
<div class='right'><a href='http://www.manchester.ac.uk/'><img src="/src/images/logos/UManchesterLogo.jpg" alt="University of Manchester" height="85" /></a></div>

Please join us for the [next GalaxyAdmins meetup on June 18](/src/community/galaxy-admins/meetups/2015-06-18/index.md) when **[Peter Briggs](http://galacticengineer.blogspot.co.uk/)** of the [University of Manchester](http://www.manchester.ac.uk/) will speak on 

 **Galactic Engineering: Experiences deploying Galaxy and developing tools within the Bioinformatics Core Facility at the University of Manchester**

[GalaxyAdmins](/src/community/galaxy-admins/index.md) will also have an [in-person meetup at GCC2015](/src/events/gcc2015/bofs/galaxy-admins/index.md).

[GalaxyAdmins](/src/community/galaxy-admins/index.md) is a special interest group for Galaxy community members who are responsible for Galaxy installations.  

## Other Events

<div class='center'> 
<a href='/src/events/asms2015/index.md'><img src="/src/images/logos/ASMSLogo.png" alt="63rd ASMS Conference on Mass Spectrometry and Allied Topics" height="80" /></a>
<a href='http://migale.jouy.inra.fr/?q=formations'><img src="/src/images/logos/MIGALELogo.png" alt="Cycle "Bioinformatique par la pratique" 2015" height="70" /></a> 
<a href='http://www.qfab.org/event/workshop-rna-seq-analysis-using-galaxy-2/'><img src="/src/images/logos/QFABLogo.png" alt="Workshop: RNA-Seq analysis using Galaxy" height="80" /></a> &nbsp;
<a href='/src/community/galaxy-admins/meetups/2015-06-18/index.md'><img src="/src/images/logos/GalaxyAdmins.png" alt="18 June GalaxyAdmins Web Meetup" height="56" /></a> &nbsp;
&nbsp;&nbsp; <a href='http://gcc2015.tsl.ac.uk/'><img src="/src/images/logos/GCC2015LogoSimple300.png" alt="2015 Galaxy Community Conference (GCC2015)" height="90" /></a>
</div>

<div class='right'><a href='http://bit.ly/gxycal'><img src="/src/images/icons/CalendarIcon.gif" /></a></div>

There are [upcoming events](/src/events/index.md) in seven countries on three continents.  See the [Galaxy Events Google Calendar](http://bit.ly/gxycal) for details on other events of interest to the community.

<table>
  <tr class="th" >
    <th> Date </th>
    <th> Topic/Event </th>
    <th> Venue/Location </th>
    <th> Contact </th>
  </tr>
  <tr>
    <th> May 31 - June 4 </th>
    <td> <strong><a href='/src/events/asms2015/index.md'>63rd ASMS Conference on Mass Spectrometry and Allied Topics</a></strong><div class='indent'><a href='/src/events/asms2015/index.md'>At least 4 talks and 5 posters</a></div> </td>
    <td> <div class='right'><img src="/src/images/icons/NA.png" alt="" /></div> St. Louis, Missouri, United States </td>
    <td> <a href='/src/events/asms2015/index.md'>Speaker List</a> </td>
  </tr>
  <tr>
    <th> June 2 </th>
    <td> <em><a href='http://migale.jouy.inra.fr/sites/all/downloads/Migale/Formations/2015/module17-2juin.pdf'>Initiation à l’utilisation de Galaxy</a></em>  </td>
    <td rowspan=3 style=" text-align: left;"> <div class='right'><img src="/src/images/icons/EU.png" alt="" /></div> <a href='http://www.jouy.inra.fr/'>Jouy-en-Josas</a>, France <div class='indent'>Partie du <a href='http://migale.jouy.inra.fr/?q=formations'>Cycle "Bioinformatique par la pratique" 2015</a></div> </td>
    <td rowspan=3> <a href="mailto:veronique.martin@jouy.inra.fr">Veronique Martin</a>, <a href="mailto:sophie.schbath@jouy.inra.fr">Sophie Schbath</a> </td>
  </tr>
  <tr>
    <th> June 3 </th>
    <td> <em><a href='http://migale.jouy.inra.fr/sites/all/downloads/Migale/Formations/2015/module8bis-3juin.pdf'>Analyse primaire de données issues de séquenceurs nouvelle génération sous Galaxy</a></em>  </td>
  </tr>
  <tr>
    <th> June 4 </th>
    <td> <em><a href='http://migale.jouy.inra.fr/sites/all/downloads/Migale/Formations/2015/module18.pdf'>Traitement bioinformatique des données RNA-Seq sous Galaxy</a></em>  </td>
  </tr>
  <tr>
    <th> June 3 </th>
    <td> <em><a href='http://www.qfab.org/event/workshop-rna-seq-analysis-using-galaxy-2/'>Workshop: RNA-Seq analysis using Galaxy</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/AU.png" alt="" /></div> <a href='http://www.qfab.org/'>QFAB</a>, University of Queensland, St Lucia, Australia </td>
    <td> <div class='right'><a href='https://training.galaxyproject.org/'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></a></div> <a href='http://www.qfab.org/mark-crowe/'>Mark Crowe</a> </td>
  </tr>
  <tr>
    <th> June 6-9 </th>
    <td> <em><a href='https://www.eshg.org/684.0.html'>WS02: NGS Variant Analysis with Galaxy</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/EU.png" alt="" /></div> <a href='https://www.eshg.org/eshg2015.0.html'>ESHG 2015</a>, Glasgow, United Kingdom </td>
    <td> <div class='right'><a href='https://training.galaxyproject.org/'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></a></div> <a href='/src/people/dave-clements/index.md'>Dave Clements</a> </td>
  </tr>
  <tr>
    <th> June 8-9 </th>
    <td> <em><a href='/src/events/glasgow2015/index.md'>Introduction to Galaxy Workshop</a></em> <div class='indent'>Workshop is full, but there is a waiting list</div> </td>
    <td> <div class='right'><img src="/src/images/icons/EU.png" alt="" /></div> <a href='http://www.gla.ac.uk/'>University of Glasgow</a>, Glasgow, United Kingdom </td>
    <td> <div class='right'><a href='https://training.galaxyproject.org/'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></a></div> <a href='/src/people/dave-clements/index.md'>Dave Clements</a> </td>
  </tr>
  <tr>
    <th> June 18 </th>
    <td> <em><a href='/src/community/galaxy-admins/meetups/2015-06-18/index.md'>Galactic Engineering: Experiences deploying Galaxy and developing tools within the Bioinformatics Core Facility at the University of Manchester</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/AllContinents40.png" alt="" /></div> <a href='/src/community/galaxy-admins/meetups/2015-06-18/index.md'>June 2015 GalaxyAdmins Web Meetup</a>, <em>Online</em> </td>
    <td> <a href='http://galacticengineer.blogspot.co.uk/'>Peter Briggs</a>, <a href='/src/people/hansrudolf-hotz/index.md'>Hans-Rudolf Hotz</a> </td>
  </tr>
  <tr>
    <th> June 22 </th>
    <td> <em>Introduction to variant calling for pathologists and laboratory managers</em> </td>
    <td> <div class='right'><img src="/src/images/icons/AU.png" alt="" /></div> Part of <a href='http://www.rcpa.edu.au/Events/Event-Details/2015/June/Short-Course-in-Medical-Genetics-and-Genetic-Patho'>Short Course in Medical Genetics and Genetic Pathology 2015</a>, Sydney, Australia </td>
    <td> <div class='right'><a href='https://training.galaxyproject.org/'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></a></div> <a href='/src/people/fubar/index.md'>Ross Lazarus</a>, <a href='https://www.vlsci.org.au/researcher/alonie'>Andrew Lonie</a> </td>
  </tr>
  <tr>
    <th> June 22 </th>
    <td> <em><a href='http://go4bioinformatics.genouest.org/initiation-a-la-plateforme-web-danalyse-de-donnees-galaxy-lanalyse-de-donnees-biologiques-pour-tous/'>Initiation à la plateforme web GALAXY</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/EU.png" alt="" /></div><a href='https://www.e-biogenouest.org/groups/guggo'>Galaxy User Group Grand Ouest (GUGGO), Angers, France</a> </td>
    <td> <div class='right'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></div> <a href='https://www.e-biogenouest.org/members/1003'>Yvan Le Bras, Cyril Monjeaud</a> </td>
  </tr>
  <tr>
    <th> June 22 </th>
    <td> <em><a href='http://go4bioinformatics.genouest.org/analyse-de-donnees-rnaseq-sous-galaxy/'>Introduction à l'analyse de données RNAseq sous Galaxy</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/EU.png" alt="" /></div><a href='https://www.e-biogenouest.org/groups/guggo'>Galaxy User Group Grand Ouest (GUGGO), Angers, France</a> </td>
    <td> <div class='right'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></div><a href='https://www.e-biogenouest.org/members/1003'>Yvan Le Bras, Cyril Monjeaud</a> </td>
  </tr>
  <tr>
    <th> June 25 </th>
    <td> <em><a href='http://go4bioinformatics.genouest.org/analyse-de-donnees-de-type-rad-par-le-pipeline-stacks-sous-la-plateforme-web-danalyse-de-donnees-galaxy/'>Initiation à l'analyse de données de type RAD par le pipeline STACKS sous Galaxy</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/EU.png" alt="" /></div>Lille, France </td>
    <td> <div class='right'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></div><a href='https://www.e-biogenouest.org/members/1003'>Yvan Le Bras, Anthony Bretaudeau, Cyril Monjeaud</a> </td>
  </tr>
  <tr>
    <th> </th>
    <td colspan=3 style=" background-color: #eef;"> </td>
  </tr>
  <tr>
    <th> July 4-8 </th>
    <td> <strong><a href='http://gcc2015.tsl.ac.uk/'>2015 Galaxy Community Conference (GCC2015)</a></strong> </td>
    <td rowspan=6> <div class='right'><img src="/src/images/icons/EU.png" alt="" /></div> <a href='http://www.tsl.ac.uk/'>The Sainsbury Lab</a>, <a href='http://www.visitnorwich.co.uk/'>Norwich</a>, United Kingdom </td>
    <td> GCC2015 Organisers </td>
  </tr>
  <tr>
    <th rowspan=2> July 4-5 </th>
    <td> <div class='indent'><strong><a href='http://gcc2015.tsl.ac.uk/data_hackathon'>Data Wrangling Hackathon</a></div></strong> </td>
    <td> Data Hackathon Organisers </td>
  </tr>
  <tr>
    <td> <div class='indent'><strong><a href='http://gcc2015.tsl.ac.uk/hackathon'>Coding Hackathon</a></div></strong> </td>
    <td> Coding Hackathon Organisers </td>
  </tr>
  <tr>
    <th> July 5 </th>
    <td> <div class='indent'><strong><a href='http://gcc2015.tsl.ac.uk/training_day'>Training SunDay</a></div></strong> </td>
    <td rowspan=2> <div class='right'><a href='https://training.galaxyproject.org/'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></a></div> GCC2015 Organisers </td>
  </tr>
  <tr>
    <th> July 6 </th>
    <td> <div class='indent'><strong><a href='http://gcc2015.tsl.ac.uk/training_day'>Training (Mon)Day</a></div></strong> </td>
  </tr>
  <tr>
    <th> July 6-8 </th>
    <td> <div class='indent'><strong><a href='http://gcc2015.tsl.ac.uk/'>GCC2015 Community Meeting</a></strong></div> </td>
    <td> GCC2015 Organisers </td>
  </tr>
  <tr>
    <th> July 6-10 </th>
    <td> <em><a href='http://bioinformatics.org.au/ws15/wp-content/uploads/ws14/sites/9/2015/02/Galaxy-Workshop-Information-Sheet.pdf'>An Introduction to Galaxy with the Genomics Virtual Lab: workshop</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/AU.png" alt="" /></div><a href='http://bioinformatics.org.au/ws15/registration/'>Winter School in Mathematical and Computational Biology</a>, University of Queensland, Brisbane, Queensland, Australia </td>
    <td> <div class='right'><a href='https://training.galaxyproject.org/'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></a></div> <a href='http://www.qfab.org/mark-crowe/'>Mark Crowe</a> </td>
  </tr>
  <tr>
    <th> July 10-14 </th>
    <td> <strong><a href='/src/events/ismb2015/index.md'>BOSC 2015 & ISMB/ECCB 2015</a></strong> <div class='indent'>10+ Galaxy related talks, posters, panel discussions, ...</div> </td>
    <td> <div class='right'><img src="/src/images/icons/EU.png" alt="" /></div> Dublin, Ireland </td>
    <td> <a href='/src/events/ismb2015/index.md'>See presenters</a> </td>
  </tr>
  <tr>
    <th> </th>
    <td colspan=3 style=" background-color: #eef;"> </td>
  </tr>
  <tr>
    <th> August 6-7 </th>
    <td> <em><a href='http://www.qfab.org/event/workshop-extended-rna-seq-analysis-3/'>Workshop: Extended RNA-Seq analysis</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/AU.png" alt="" /></div> <a href='http://www.qfab.org/'>QFAB</a>, University of Queensland, St Lucia, Australia </td>
    <td> <div class='right'><a href='https://training.galaxyproject.org/'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></a></div> <a href='http://www.qfab.org/mark-crowe/'>Mark Crowe</a> </td>
  </tr>
  <tr>
    <th> </th>
    <td colspan=3 style=" background-color: #eef;"> </td>
  </tr>
  <tr>
    <th> September<br />4, 11 </th>
    <td> <em><a href='http://www.vib.be/en/training/research-training/courses/Pages/Introduction-to-NGS-data-analysis.aspx'>Introduction to NGS data analysis</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/EU.png" alt="" /></div> Leuven, Belgium </td>
    <td> <a href='http://www.vib.be/whoiswho/Pages/Janick-Mathys.aspx?lang=en'>Janick Mathys</a> </td>
  </tr>
  <tr>
    <th> September 14-18 </th>
    <td> <em><a href='http://training.bioinformatics.ucdavis.edu/2015/01/13/using-galaxy-for-analysis-of-high-throughput-sequence-data-september-14-18-2015/'>Using Galaxy for Analysis of High Throughput Sequence Data, September</a></em> </td>
    <td> <div class='right'><img src="/src/images/icons/NA.png" alt="" /></div> <a href='http://bioinformatics.ucdavis.edu/'>UC Davis Bioinformatics Core</a>, Davis, California, United States </td>
    <td> <div class='right'><a href='http://bit.ly/gxytrnUCDavis'><img src="/src/images/icons/GTN32.png" alt="Training offered by GTN Member" width="16" /></a></div> <a href="mailto:bioinformatics.core@ucdavis.edu">UC Davis Bioinformatics</a> </td>
  </tr>
</table>


<table>
  <tr>
    <td style=" border: none;"> <a href='https://training.galaxyproject.org/'><img src="/src/images/icons/GTN32.png" alt="Designates a training event offered by GTN Member" width="24" /></a> </td>
    <td style=" border: none;"> <span style="font-size: smaller;"> <a href='https://training.galaxyproject.org/'>Designates a training event offered by GTN member(s)</a> </span> </td>
  </tr>
</table>


----

# Who's Hiring

<div class='right'><a href='/src/galaxy-is-hiring/index.md'><img src="/src/galaxy-is-hiring/GalaxyIsHiringWordCloud2.png" alt="Please Help! Yes you!" width="200" /></a></div>

The Galaxy is expanding! Please help it grow.

* [Post-Doc CDD-IR CEA-Institut de Génomique](http://bit.ly/1Pbd9gk), Evry, France 
    Application deadline extended to 2015/06/30
* [Computational Metabolomics Professor](https://psu.jobs/job/55363), Penn State University, Pennsylvania, United States
* **The [Galaxy Project is hiring software engineers and post-docs](/src/galaxy-is-hiring/index.md)**

Got a Galaxy-related opening? Send it to outreach@galaxyproject.org and we'll put it in the [Galaxy News feed](/src/news/index.md) and include it in next month's [update](/src/galaxy-updates/index.md).

<br />

----

# New Public Galaxy Servers

The [VirAmp](http://viramp.com/) server was added to the [list of public Galaxy servers](/src/use/index.md) in May:

## VirAmp

<div class='right solid'><a href='http://viramp.com/'><img src="/src/use/viramp/viramp-home.png" alt="VirAmp" width=300 /></a></div>

**[VirAmp](http://viramp.com/)** is a web-based semi-*de novo* fast virus genome assembly pipeline designed for extremely high coverage NGS data. From [VirAmp: a galaxy-based viral genome assembly pipeline](http://www.gigasciencejournal.com/content/4/1/19), Yinan Wan, Daniel W Renner, Istvan Albert and Moriah L Szpara, *[GigaScience](http://www.gigasciencejournal.com/)* 2015, 4:19 doi:10.1186/s13742-015-0060-y
      [VirAmp](http://viramp.com/) "combines existing tools and techniques and presents them to end users via a web-enabled Galaxy interface. Our pipeline allows users to assemble, analyze, and interpret high coverage viral sequencing data with an ease and efficiency that was not possible previously. Our software makes a large number of genome assembly and related tools available to life scientists and automates the currently recommended best practices into a single, easy to use interface. We tested our pipeline with three different datasets from human herpes simplex virus (HSV)."

VirAmp includes [Documentation](http://docs.viramp.com/en/latest/) and is supported by the [Huck Institutes of the Life Sciences](https://www.huck.psu.edu/) and the [Department of Biochemistry and Molecular Biology](http://bmb.psu.edu/), and [Pennsylvania State University](http://www.psu.edu/)



----

# Releases

## May 2015 Galaxy Release (v 15.05)

<div class='right'><a href='http://getgalaxy.org'><img src="/src/images/logos/GalaxyNewLogo_GalaxyProject_Trans.png" alt="GalaxyProject" width="200" /></a></div>
 
**The May 2015 Galaxy Release is out!**
**[Release Notes v 15.05](http://galaxy.readthedocs.org/en/master/releases/15.05_announce.html)**

**Release Highlights**

* **Authentication Plugins** - Galaxy now has native support for LDAP and Active Directory via a new community developed authentication plugin system.
* **Tool Sections** - Tool parameters may now be groupped into collapsable sections.
* **Collection Creators** - New widgets have been added that allow much more flexibility when creating simple dataset pair and list collections.
* And *[much more](http://galaxy.readthedocs.org/en/master/releases/15.05_announce.html)*

The release notes have a new home at **[Read the Docs](http://galaxy.readthedocs.org/en/master/releases/15.05_announce.html)**. We hope that you like the new format as much as we do!

*Thanks for using Galaxy!* <br />
[The Galaxy Team](/src/galaxy-team/index.md)

## Pulsar 0.5.0

[Pulsar 0.5.0](https://pypi.python.org/pypi/pulsar-app/0.5.0) was released in May.  Enhancements include:

* Allow cURL downloader to resume transfers during staging in (thanks to [Nate Coraor](https://github.com/natefoo)). [0c61bd9](https://github.com/galaxyproject/pulsar/commit/0c61bd9)
* Fix to cURL downloaders status code handling (thanks to [Nate Coraor](https://github.com/natefoo)). [86f95ce](https://github.com/galaxyproject/pulsar/commit/86f95ce)
* Fix non-wheel installs from PyPI. [Issue 72](https://github.com/galaxyproject/pulsar/issues/72)
* Fix mesos imports for newer versions of mesos (thanks to [Kyle Ellrott](https://github.com/kellrott)). [fe3e919](https://github.com/galaxyproject/pulsar/commit/fe3e919)
* More, better logging. [2b3942d](https://github.com/galaxyproject/pulsar/commit/2b3942d), [fa2b6dc](https://github.com/galaxyproject/pulsar/commit/fa2b6dc)

Pulsar is a Python server application that allows a Galaxy server to run jobs on remote systems (including Windows) without requiring a shared mounted file systems. Unlike traditional Galaxy job runners - input files, scripts, and config files may be transferred to the remote system, the job is executed, and the results are transferred back to the Galaxy server - eliminating the need for a shared file system.

## Planemo 0.12.2

<div class='left'><a href='https://pypi.python.org/pypi/planemo/'><img src="/src/images/logos/PlanemoLogo.png" alt="" width="200" /></a></div>

Planemo saw [four major and several minor](https://pypi.python.org/pypi/planemo/0.12.2#history) releases this past month.  The most recent is [v0.12.2](https://pypi.python.org/pypi/planemo/0.12.2).  See the [release history](http://planemo.readthedocs.org/en/latest/history.html#id2) for full details and links.

Planemo is a set of command-line utilities to assist in building tools for the Galaxy project.


## Others

 **[CloudMan](/src/cloudman/index.md) **
   The [most recent edition of CloudMan](/src/galaxy-updates/2014-09/index.md#august-2014-cloudman-release) was released in August.

 **BioBlend V0.5.3**
   [BioBlend v0.3](/src/news/bio-blend-v0.5.3/index.md) was also released in March.<br />
   [BioBlend](https://github.com/galaxyproject/bioblend/blob) is a python library for interacting with CloudMan and the [Galaxy API](/src/learn/api/index.md).  ([CloudMan](/src/cloudman/index.md) offers an easy way to get a personal and completely functional instance of Galaxy in the cloud in just a few minutes, without any manual configuration.)

 **blend4j v0.1.2**
   [blend4j v0.1.2](https://github.com/jmchilton/blend4j) was released in December 2014.  blend4j is a JVM partial reimplemenation of the Python library bioblend for interacting with Galaxy, CloudMan, and BioCloudCentral.


----

# Other News

* From [John Chilton](https://twitter.com/jmchilton):
  * EDAM ([edamontology.org](http://edamontology.org)) format added to all Galaxy datatypes. Huge thanks to Pasteur team & Helena Rasche!
  * [Connecting a dockerized Galaxy container to a Condor cluster](https://registry.hub.docker.com/u/dpryan79/htcondor-galaxy/) pretty awesome stuff from [Devon Ryan](https://twitter.com/dpryan79).
  * Just merged [collaborative work](https://github.com/galaxyproject/galaxy/pull/230) spearheaded by Helena Rasche adding RStudio integration to Galaxy
* [Videos from Pitagora Galaxy workshop](http://bit.ly/1CPXf8v) are now online
* [GigaScience Pushes Metabolomics Open Data & Training](http://blogs.biomedcentral.com/gigablog/2015/05/13/gigascience-pushes-metabolomics-open-data-training/), by [Scott Edmunds](http://blogs.biomedcentral.com/gigablog/author/scottedmunds)

<div class='right'><br /><a href='http://toolshed.g2.bx.psu.edu/'><img src="/src/images/logos/ToolShed.jpg" alt="Galaxy ToolShed" width=150 /></a></div>

# ToolShed Contributions

See [list of tools contributed in May](/src/toolshed/contributions/2015-05/index.md).
