---
title: "MiModD NacreousMap"
url: "http://mapping-by-sequencing.vm.uni-freiburg.de:8080/"
scope: "tool-publishing"
platforms:
  - platform_group: "public-server"
    platform_url: "http://mapping-by-sequencing.vm.uni-freiburg.de:8080/"
    platform_text: "MiModD NacreousMap"
    platform_location: DE
summary: "NacreousMap mapping/plotting tool of MiModD for users of MiModD who do not want to install the required dependencies (R and rpy2) for graphical output from this tool on their local system. "
image: "/src/use/mimodd-nacreousmap/baumeister.png"
comments:
  - "[MiModD](http://www.celegans.de/en/mimodd) is a comprehensive software package for the identification and annotation of mutations in the genomes of model organisms from whole-genome sequencing (WGS) data."
  - "[CloudMap](https://usegalaxy.org/cloudmap) users can replot data produced with the Hawaiian Variant Mapping tool using the NacreousMap engine to obtain optimized (much smaller files that display faster) plots."
  - "To install the complete MiModD package for use as a command line tool or for integration into any local Galaxy follow the [installation instructions](http://mimodd.readthedocs.org/en/latest/INSTALL.html) in the [MiModD User Guide](http://mimodd.readthedocs.org/en/latest/)."
user_support:
  - "[MiModD User Guide](http://mimodd.readthedocs.org/en/latest/index.html#)"
  - "[MiModD Google Group](mailto:mimodd@googlegroups.com)"
quotas:
  - "The quota for anonymous usage is 50MB, registered users have 250MB available."
  - 'You can analyze/plot data in VCF format or the "Per variant report" format generated by local runs of the MiModD NacreousMap tool. The latter has the advantage of being up to 20 times smaller than the corresponding VCF file.'
citations:
pub_libraries:
  - "MiModD"
sponsors:
  - "[The Baumeister Laboratory](http://www.celegans.de/) at the [Institute for Biology 3](http://www.biologie.uni-freiburg.de/data/bio3/) of the [Albert-Ludwig University](http://www.uni-freiburg.de/) in Freiburg/Breisgau, Germany."
---
